{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"Copy of Student_DNADetectives_Bonus_Section5.ipynb","provenance":[{"file_id":"1kwhCbWWeiN_S5aHo2qZVB0rD_P0ZzPCV","timestamp":1656521093579}],"collapsed_sections":["zzpfWAciDqG9"]},"kernelspec":{"name":"python3","display_name":"Python 3"}},"cells":[{"cell_type":"markdown","metadata":{"id":"qLlU8RKtpy52"},"source":["# Goals: \n","In this colab you will:\n","- Learn what DNA sequencing is, and why we care about it.\n","- Understand what the output from next generation sequencing looks like.\n","- Understand the difference between *de novo* assembly and reference based alignment.\n","- Perform your own *de novo* assembly of the SARS-CoV-2 genome!\n","- Perform your own reference-based alignment of the SARS-CoV-2 genome!\n","- View and understand your alignment using IGVviewer.\n","\n","**Note: This colab does not use any machine learning, but rather teaches some fundamental concepts of genomics/bioinformatics. It is intended for those who are interested in using the state-of-the-art technologies that graduate students, professors, and companies use to analyze DNA.**"]},{"cell_type":"code","metadata":{"id":"pXFCoQqmVr3H","cellView":"form","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1656521199025,"user_tz":240,"elapsed":58056,"user":{"displayName":"Arav Bhardwaj","userId":"05330003763859094861"}},"outputId":"0125dfc5-f081-4f16-e390-48cb03c09d7d"},"source":["#@title # Install libraries - ignore warnings/errors.\n","%%bash \n","\n","# Make a local/temp directory called 'software' and download the needed software into it.\n","mkdir software\n","mkdir data\n","mkdir results\n","cd software\n","\n","# Download/install required software packages (bwa, SPAdes, and samtools) -- \n","# this just installs it to your temporary/local/cloud instance NOT to your \n","# permanent gDrive.\n","git clone https://github.com/lh3/bwa.git\n","wget -O samtools.tar.bz2 'https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/samtools-1.3.1.tar.bz2'\n","tar -xjvf samtools.tar.bz2 \n","mv samtools-1.3.1 samtools\n","rm *.bz2\n","wget -O SPAdes-3.12.0-Linux.tar.gz 'https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/SPAdes-3.12.0-Linux.tar.gz'\n","tar -xzf SPAdes-3.12.0-Linux.tar.gz\n","mv SPAdes-3.12.0-Linux SPAdes\n","rm *.gz\n","chmod -R 777 '/content/software'\n","cd bwa; make\n","cd /content/software/samtools\n","configure; make; make install\n"],"execution_count":1,"outputs":[{"output_type":"stream","name":"stdout","text":["samtools-1.3.1/\n","samtools-1.3.1/aclocal.m4\n","samtools-1.3.1/AUTHORS\n","samtools-1.3.1/bam.c\n","samtools-1.3.1/bam.h\n","samtools-1.3.1/bam2bcf.c\n","samtools-1.3.1/bam2bcf.h\n","samtools-1.3.1/bam2bcf_indel.c\n","samtools-1.3.1/bam2depth.c\n","samtools-1.3.1/bam_addrprg.c\n","samtools-1.3.1/bam_aux.c\n","samtools-1.3.1/bam_cat.c\n","samtools-1.3.1/bam_color.c\n","samtools-1.3.1/bam_endian.h\n","samtools-1.3.1/bam_flags.c\n","samtools-1.3.1/bam_import.c\n","samtools-1.3.1/bam_index.c\n","samtools-1.3.1/bam_lpileup.c\n","samtools-1.3.1/bam_lpileup.h\n","samtools-1.3.1/bam_mate.c\n","samtools-1.3.1/bam_md.c\n","samtools-1.3.1/bam_plbuf.c\n","samtools-1.3.1/bam_plbuf.h\n","samtools-1.3.1/bam_plcmd.c\n","samtools-1.3.1/bam_quickcheck.c\n","samtools-1.3.1/bam_reheader.c\n","samtools-1.3.1/bam_rmdup.c\n","samtools-1.3.1/bam_rmdupse.c\n","samtools-1.3.1/bam_sort.c\n","samtools-1.3.1/bam_split.c\n","samtools-1.3.1/bam_stat.c\n","samtools-1.3.1/bam_tview.c\n","samtools-1.3.1/bam_tview.h\n","samtools-1.3.1/bam_tview_curses.c\n","samtools-1.3.1/bam_tview_html.c\n","samtools-1.3.1/bamshuf.c\n","samtools-1.3.1/bamtk.c\n","samtools-1.3.1/bedcov.c\n","samtools-1.3.1/bedidx.c\n","samtools-1.3.1/ChangeLog.old\n","samtools-1.3.1/config.h.in\n","samtools-1.3.1/config.mk.in\n","samtools-1.3.1/configure\n","samtools-1.3.1/configure.ac\n","samtools-1.3.1/cut_target.c\n","samtools-1.3.1/dict.c\n","samtools-1.3.1/errmod.c\n","samtools-1.3.1/errmod.h\n","samtools-1.3.1/examples/\n","samtools-1.3.1/faidx.c\n","samtools-1.3.1/htslib-1.3.1/\n","samtools-1.3.1/INSTALL\n","samtools-1.3.1/install-sh\n","samtools-1.3.1/kprobaln.c\n","samtools-1.3.1/kprobaln.h\n","samtools-1.3.1/LICENSE\n","samtools-1.3.1/Makefile\n","samtools-1.3.1/Makefile.mingw\n","samtools-1.3.1/misc/\n","samtools-1.3.1/NEWS\n","samtools-1.3.1/padding.c\n","samtools-1.3.1/phase.c\n","samtools-1.3.1/README\n","samtools-1.3.1/sam.c\n","samtools-1.3.1/sam.h\n","samtools-1.3.1/sam_header.c\n","samtools-1.3.1/sam_header.h\n","samtools-1.3.1/sam_opts.c\n","samtools-1.3.1/sam_opts.h\n","samtools-1.3.1/sam_view.c\n","samtools-1.3.1/sample.c\n","samtools-1.3.1/sample.h\n","samtools-1.3.1/samtools.1\n","samtools-1.3.1/samtools.h\n","samtools-1.3.1/stats.c\n","samtools-1.3.1/stats_isize.c\n","samtools-1.3.1/stats_isize.h\n","samtools-1.3.1/test/\n","samtools-1.3.1/win32/\n","samtools-1.3.1/win32/libcurses.a\n","samtools-1.3.1/win32/libz.a\n","samtools-1.3.1/win32/xcurses.h\n","samtools-1.3.1/win32/zconf.h\n","samtools-1.3.1/win32/zlib.h\n","samtools-1.3.1/test/addrprg/\n","samtools-1.3.1/test/bam2fq/\n","samtools-1.3.1/test/dat/\n","samtools-1.3.1/test/fixmate/\n","samtools-1.3.1/test/idxstats/\n","samtools-1.3.1/test/merge/\n","samtools-1.3.1/test/mpileup/\n","samtools-1.3.1/test/quickcheck/\n","samtools-1.3.1/test/reheader/\n","samtools-1.3.1/test/split/\n","samtools-1.3.1/test/stat/\n","samtools-1.3.1/test/test.c\n","samtools-1.3.1/test/test.h\n","samtools-1.3.1/test/test.pl\n","samtools-1.3.1/test/tview/\n","samtools-1.3.1/test/vcf-miniview.c\n","samtools-1.3.1/test/tview/test_get_rg_sample.c\n","samtools-1.3.1/test/stat/1.stats.expected\n","samtools-1.3.1/test/stat/10.stats.expected\n","samtools-1.3.1/test/stat/10_map_cigar.sam\n","samtools-1.3.1/test/stat/10_map_cigar.sam_s1_a_1.expected.bamstat\n","samtools-1.3.1/test/stat/10_map_cigar.sam_s1_b_1.expected.bamstat\n","samtools-1.3.1/test/stat/1_map_cigar.sam\n","samtools-1.3.1/test/stat/1_map_cigar.sam_s1_a_1.expected.bamstat\n","samtools-1.3.1/test/stat/2.stats.expected\n","samtools-1.3.1/test/stat/2_equal_cigar_full_seq.sam\n","samtools-1.3.1/test/stat/3.stats.expected\n","samtools-1.3.1/test/stat/3_map_cigar_equal_seq.sam\n","samtools-1.3.1/test/stat/4.stats.expected\n","samtools-1.3.1/test/stat/4_X_cigar_full_seq.sam\n","samtools-1.3.1/test/stat/5.stats.expected\n","samtools-1.3.1/test/stat/5_insert_cigar.sam\n","samtools-1.3.1/test/stat/6.stats.expected\n","samtools-1.3.1/test/stat/7.stats.expected\n","samtools-1.3.1/test/stat/7_supp.sam\n","samtools-1.3.1/test/stat/8.stats.expected\n","samtools-1.3.1/test/stat/8_secondary.sam\n","samtools-1.3.1/test/stat/9.stats.expected\n","samtools-1.3.1/test/stat/test.fa\n","samtools-1.3.1/test/stat/test.fa.fai\n","samtools-1.3.1/test/split/test_count_rg.c\n","samtools-1.3.1/test/split/test_expand_format_string.c\n","samtools-1.3.1/test/split/test_filter_header_rg.c\n","samtools-1.3.1/test/split/test_parse_args.c\n","samtools-1.3.1/test/reheader/1_view1.sam.expected\n","samtools-1.3.1/test/reheader/1_view1.sam.expected.err\n","samtools-1.3.1/test/reheader/2_view1.sam.expected\n","samtools-1.3.1/test/reheader/2_view1.sam.expected.err\n","samtools-1.3.1/test/reheader/3_view1.sam.expected\n","samtools-1.3.1/test/reheader/3_view1.sam.expected.err\n","samtools-1.3.1/test/reheader/hdr.sam\n","samtools-1.3.1/test/quickcheck/1.quickcheck.badeof.bam\n","samtools-1.3.1/test/quickcheck/2.quickcheck.badheader.bam\n","samtools-1.3.1/test/quickcheck/3.quickcheck.ok.bam\n","samtools-1.3.1/test/quickcheck/4.quickcheck.ok.bam\n","samtools-1.3.1/test/quickcheck/5.quickcheck.truncated.cram\n","samtools-1.3.1/test/quickcheck/all.expected\n","samtools-1.3.1/test/mpileup/1read.bam\n","samtools-1.3.1/test/mpileup/1read.sam\n","samtools-1.3.1/test/mpileup/anomalous.sam\n","samtools-1.3.1/test/mpileup/c1#clip.bam\n","samtools-1.3.1/test/mpileup/c1#ID.bam\n","samtools-1.3.1/test/mpileup/c1#ID2.bam\n","samtools-1.3.1/test/mpileup/c1#pad1.bam\n","samtools-1.3.1/test/mpileup/c1#pad2.bam\n","samtools-1.3.1/test/mpileup/c1#pad3.bam\n","samtools-1.3.1/test/mpileup/c1.fa\n","samtools-1.3.1/test/mpileup/c1.fa.fai\n","samtools-1.3.1/test/mpileup/ce#5b.bam\n","samtools-1.3.1/test/mpileup/ce#5b.bam.bai\n","samtools-1.3.1/test/mpileup/ce#5b.cram\n","samtools-1.3.1/test/mpileup/ce#5b.cram.crai\n","samtools-1.3.1/test/mpileup/ce#5b.sam\n","samtools-1.3.1/test/mpileup/ce#large_seq.bam\n","samtools-1.3.1/test/mpileup/ce#unmap.bam\n","samtools-1.3.1/test/mpileup/ce#unmap1.bam\n","samtools-1.3.1/test/mpileup/ce#unmap2.bam\n","samtools-1.3.1/test/mpileup/ce.fa\n","samtools-1.3.1/test/mpileup/ce.fa.fai\n","samtools-1.3.1/test/mpileup/deep.sam\n","samtools-1.3.1/test/mpileup/depth.reg\n","samtools-1.3.1/test/mpileup/expected/\n","samtools-1.3.1/test/mpileup/indels.sam\n","samtools-1.3.1/test/mpileup/md5/\n","samtools-1.3.1/test/mpileup/mpileup-E.bam\n","samtools-1.3.1/test/mpileup/mpileup.1.bam\n","samtools-1.3.1/test/mpileup/mpileup.1.bam.bai\n","samtools-1.3.1/test/mpileup/mpileup.2.bam\n","samtools-1.3.1/test/mpileup/mpileup.3.bam\n","samtools-1.3.1/test/mpileup/mpileup.ref.fa\n","samtools-1.3.1/test/mpileup/mpileup.ref.fa.fai\n","samtools-1.3.1/test/mpileup/mpileup.reg\n","samtools-1.3.1/test/mpileup/overlap50.sam\n","samtools-1.3.1/test/mpileup/overlapIllumina.bam\n","samtools-1.3.1/test/mpileup/regions\n","samtools-1.3.1/test/mpileup/regression.sh\n","samtools-1.3.1/test/mpileup/xx#depth1.sam\n","samtools-1.3.1/test/mpileup/xx#depth2.sam\n","samtools-1.3.1/test/mpileup/xx#minimal.bam\n","samtools-1.3.1/test/mpileup/xx#triplet.bam\n","samtools-1.3.1/test/mpileup/xx.bed\n","samtools-1.3.1/test/mpileup/xx.bed2\n","samtools-1.3.1/test/mpileup/xx.fa\n","samtools-1.3.1/test/mpileup/xx.fa.fai\n","samtools-1.3.1/test/mpileup/md5/02099b2aad2cfe707759353724fb40ae\n","samtools-1.3.1/test/mpileup/md5/0d4dfc45807d6a11439e858605cb364d\n","samtools-1.3.1/test/mpileup/md5/351f64d4f4f9ddd45b35336ad97aa6de\n","samtools-1.3.1/test/mpileup/md5/6ec590eb82c361a61e6baabce20109bc\n","samtools-1.3.1/test/mpileup/md5/6f60cdf4c739651ae72d53e2e94825c1\n","samtools-1.3.1/test/mpileup/md5/73d180d5cb3860a2713927b6812eab0e\n","samtools-1.3.1/test/mpileup/expected/1.out.f3-6.gz\n","samtools-1.3.1/test/mpileup/expected/10.out\n","samtools-1.3.1/test/mpileup/expected/11.out\n","samtools-1.3.1/test/mpileup/expected/12.out\n","samtools-1.3.1/test/mpileup/expected/13.out\n","samtools-1.3.1/test/mpileup/expected/14.out\n","samtools-1.3.1/test/mpileup/expected/15.out\n","samtools-1.3.1/test/mpileup/expected/16.out\n","samtools-1.3.1/test/mpileup/expected/17.out\n","samtools-1.3.1/test/mpileup/expected/18.out\n","samtools-1.3.1/test/mpileup/expected/19.out\n","samtools-1.3.1/test/mpileup/expected/2.out\n","samtools-1.3.1/test/mpileup/expected/20.out\n","samtools-1.3.1/test/mpileup/expected/21.out\n","samtools-1.3.1/test/mpileup/expected/22.out\n","samtools-1.3.1/test/mpileup/expected/23.out\n","samtools-1.3.1/test/mpileup/expected/24.out\n","samtools-1.3.1/test/mpileup/expected/25.out\n","samtools-1.3.1/test/mpileup/expected/26.out\n","samtools-1.3.1/test/mpileup/expected/27.out\n","samtools-1.3.1/test/mpileup/expected/28.out\n","samtools-1.3.1/test/mpileup/expected/29.out\n","samtools-1.3.1/test/mpileup/expected/3.out\n","samtools-1.3.1/test/mpileup/expected/30.out\n","samtools-1.3.1/test/mpileup/expected/31.out\n","samtools-1.3.1/test/mpileup/expected/32.out\n","samtools-1.3.1/test/mpileup/expected/33.out\n","samtools-1.3.1/test/mpileup/expected/34.out\n","samtools-1.3.1/test/mpileup/expected/35.out\n","samtools-1.3.1/test/mpileup/expected/37.out\n","samtools-1.3.1/test/mpileup/expected/38.out\n","samtools-1.3.1/test/mpileup/expected/39.out\n","samtools-1.3.1/test/mpileup/expected/4.out\n","samtools-1.3.1/test/mpileup/expected/40.out\n","samtools-1.3.1/test/mpileup/expected/41.out\n","samtools-1.3.1/test/mpileup/expected/42.out\n","samtools-1.3.1/test/mpileup/expected/43.out\n","samtools-1.3.1/test/mpileup/expected/44.out\n","samtools-1.3.1/test/mpileup/expected/45.out\n","samtools-1.3.1/test/mpileup/expected/46.out\n","samtools-1.3.1/test/mpileup/expected/47.out\n","samtools-1.3.1/test/mpileup/expected/48.out\n","samtools-1.3.1/test/mpileup/expected/49.out\n","samtools-1.3.1/test/mpileup/expected/5.out\n","samtools-1.3.1/test/mpileup/expected/50.out\n","samtools-1.3.1/test/mpileup/expected/51.out\n","samtools-1.3.1/test/mpileup/expected/52.out\n","samtools-1.3.1/test/mpileup/expected/53.out\n","samtools-1.3.1/test/mpileup/expected/54.out\n","samtools-1.3.1/test/mpileup/expected/55.out\n","samtools-1.3.1/test/mpileup/expected/56.out\n","samtools-1.3.1/test/mpileup/expected/57.out\n","samtools-1.3.1/test/mpileup/expected/58.out\n","samtools-1.3.1/test/mpileup/expected/59.out\n","samtools-1.3.1/test/mpileup/expected/6.out\n","samtools-1.3.1/test/mpileup/expected/60.out\n","samtools-1.3.1/test/mpileup/expected/61.out\n","samtools-1.3.1/test/mpileup/expected/62.out\n","samtools-1.3.1/test/mpileup/expected/63.out\n","samtools-1.3.1/test/mpileup/expected/64.out\n","samtools-1.3.1/test/mpileup/expected/65.out\n","samtools-1.3.1/test/mpileup/expected/66.out\n","samtools-1.3.1/test/mpileup/expected/67.out\n","samtools-1.3.1/test/mpileup/expected/68.out\n","samtools-1.3.1/test/mpileup/expected/69.out\n","samtools-1.3.1/test/mpileup/expected/7.out\n","samtools-1.3.1/test/mpileup/expected/70.out\n","samtools-1.3.1/test/mpileup/expected/71.out\n","samtools-1.3.1/test/mpileup/expected/72.out\n","samtools-1.3.1/test/mpileup/expected/73.out\n","samtools-1.3.1/test/mpileup/expected/74.out\n","samtools-1.3.1/test/mpileup/expected/75.out\n","samtools-1.3.1/test/mpileup/expected/76.out\n","samtools-1.3.1/test/mpileup/expected/77.out\n","samtools-1.3.1/test/mpileup/expected/8.out\n","samtools-1.3.1/test/mpileup/expected/9.out\n","samtools-1.3.1/test/mpileup/expected/d1_1.out\n","samtools-1.3.1/test/mpileup/expected/d1_12.out\n","samtools-1.3.1/test/mpileup/expected/d1_2.out\n","samtools-1.3.1/test/mpileup/expected/d2_12r.out\n","samtools-1.3.1/test/mpileup/expected/d3_12.out\n","samtools-1.3.1/test/mpileup/expected/d3_12r1a.out\n","samtools-1.3.1/test/mpileup/expected/d3_12r2a.out\n","samtools-1.3.1/test/mpileup/expected/d4_12.out\n","samtools-1.3.1/test/mpileup/expected/d4_12b.out\n","samtools-1.3.1/test/mpileup/expected/d4_12r.out\n","samtools-1.3.1/test/merge/2.merge.expected.bam\n","samtools-1.3.1/test/merge/3.merge.expected.bam\n","samtools-1.3.1/test/merge/3.merge.expected.err\n","samtools-1.3.1/test/merge/4.merge.expected.bam\n","samtools-1.3.1/test/merge/5.merge.expected.bam\n","samtools-1.3.1/test/merge/5.merge.expected.sam\n","samtools-1.3.1/test/merge/6.merge.expected.bam\n","samtools-1.3.1/test/merge/7.merge.expected.bam\n","samtools-1.3.1/test/merge/test_3.fofn\n","samtools-1.3.1/test/merge/test_bam_translate.c\n","samtools-1.3.1/test/merge/test_header.sam\n","samtools-1.3.1/test/merge/test_rtrans_build.c\n","samtools-1.3.1/test/merge/test_trans_tbl_init.c\n","samtools-1.3.1/test/idxstats/test_input_1_a.bam.expected\n","samtools-1.3.1/test/idxstats/test_input_1_a.bam.expected.err\n","samtools-1.3.1/test/fixmate/1_coord_sort.sam\n","samtools-1.3.1/test/fixmate/1_coord_sort.sam.expected\n","samtools-1.3.1/test/fixmate/1_coord_sort.sam.expected.err\n","samtools-1.3.1/test/fixmate/2_isize_overflow.sam\n","samtools-1.3.1/test/fixmate/2_isize_overflow.sam.expected\n","samtools-1.3.1/test/fixmate/3_reverse_read_pp_lt.sam\n","samtools-1.3.1/test/fixmate/3_reverse_read_pp_lt.sam.expected\n","samtools-1.3.1/test/fixmate/4_reverse_read_pp_equal.sam\n","samtools-1.3.1/test/fixmate/4_reverse_read_pp_equal.sam.expected\n","samtools-1.3.1/test/fixmate/5_ct.sam\n","samtools-1.3.1/test/fixmate/5_ct.sam.expected\n","samtools-1.3.1/test/fixmate/6_ct_replace.sam\n","samtools-1.3.1/test/fixmate/6_ct_replace.sam.expected\n","samtools-1.3.1/test/dat/bam2fq.001.sam\n","samtools-1.3.1/test/dat/bam2fq.002.sam\n","samtools-1.3.1/test/dat/bam2fq.003.sam\n","samtools-1.3.1/test/dat/cat.hdr\n","samtools-1.3.1/test/dat/cram_md5/\n","samtools-1.3.1/test/dat/depad.001.fa\n","samtools-1.3.1/test/dat/depad.001.fa.fai\n","samtools-1.3.1/test/dat/depad.001p.sam\n","samtools-1.3.1/test/dat/depad.001u.sam\n","samtools-1.3.1/test/dat/dict.fa\n","samtools-1.3.1/test/dat/dict.out\n","samtools-1.3.1/test/dat/empty.expected\n","samtools-1.3.1/test/dat/large_chrom.out\n","samtools-1.3.1/test/dat/large_chrom.sam\n","samtools-1.3.1/test/dat/large_chrom_bai_index.err\n","samtools-1.3.1/test/dat/mpileup.1.sam\n","samtools-1.3.1/test/dat/mpileup.2.sam\n","samtools-1.3.1/test/dat/mpileup.3.sam\n","samtools-1.3.1/test/dat/mpileup.err.1\n","samtools-1.3.1/test/dat/mpileup.out.1\n","samtools-1.3.1/test/dat/mpileup.out.2\n","samtools-1.3.1/test/dat/mpileup.out.3\n","samtools-1.3.1/test/dat/mpileup.out.4\n","samtools-1.3.1/test/dat/mpileup.ref.fa\n","samtools-1.3.1/test/dat/mpileup.ref.fa.fai\n","samtools-1.3.1/test/dat/test_input_1_a.bam\n","samtools-1.3.1/test/dat/test_input_1_a.bam.bai.expected\n","samtools-1.3.1/test/dat/test_input_1_a.sam\n","samtools-1.3.1/test/dat/test_input_1_a_regex.sam\n","samtools-1.3.1/test/dat/test_input_1_b.bam\n","samtools-1.3.1/test/dat/test_input_1_b.sam\n","samtools-1.3.1/test/dat/test_input_1_b_regex.sam\n","samtools-1.3.1/test/dat/test_input_1_c.bam\n","samtools-1.3.1/test/dat/test_input_1_c.sam\n","samtools-1.3.1/test/dat/view.001.01.bed\n","samtools-1.3.1/test/dat/view.001.02.bed\n","samtools-1.3.1/test/dat/view.001.fa\n","samtools-1.3.1/test/dat/view.001.fa.fai\n","samtools-1.3.1/test/dat/view.001.sam\n","samtools-1.3.1/test/dat/view.002.01.bed\n","samtools-1.3.1/test/dat/view.002.02.bed\n","samtools-1.3.1/test/dat/view.002.fa\n","samtools-1.3.1/test/dat/view.002.fa.fai\n","samtools-1.3.1/test/dat/view.002.sam\n","samtools-1.3.1/test/dat/view.003.expected.sam\n","samtools-1.3.1/test/dat/view.003.sam\n","samtools-1.3.1/test/dat/cram_md5/08c04d512d4797d9ba2a156c1daba468\n","samtools-1.3.1/test/dat/cram_md5/7c35feac7036c1cdef3bee0cc4b21437\n","samtools-1.3.1/test/dat/cram_md5/89b96de7acc5918c2858ef92710be7d4\n","samtools-1.3.1/test/dat/cram_md5/8a3d8deb9edd4ce3346b6061fcee6d0b\n","samtools-1.3.1/test/bam2fq/1.1.fq.expected\n","samtools-1.3.1/test/bam2fq/1.2.fq.expected\n","samtools-1.3.1/test/bam2fq/1.stdout.expected\n","samtools-1.3.1/test/bam2fq/2.1.fq.expected\n","samtools-1.3.1/test/bam2fq/2.2.fq.expected\n","samtools-1.3.1/test/bam2fq/2.s.fq.expected\n","samtools-1.3.1/test/bam2fq/2.stdout.expected\n","samtools-1.3.1/test/bam2fq/3.1.fq.expected\n","samtools-1.3.1/test/bam2fq/3.2.fq.expected\n","samtools-1.3.1/test/bam2fq/3.s.fq.expected\n","samtools-1.3.1/test/bam2fq/3.stdout.expected\n","samtools-1.3.1/test/bam2fq/4.1.fq.expected\n","samtools-1.3.1/test/bam2fq/4.2.fq.expected\n","samtools-1.3.1/test/bam2fq/4.s.fq.expected\n","samtools-1.3.1/test/bam2fq/4.stdout.expected\n","samtools-1.3.1/test/addrprg/1_fixup.sam\n","samtools-1.3.1/test/addrprg/1_fixup.sam.expected\n","samtools-1.3.1/test/addrprg/1_fixup.sam.expected.err\n","samtools-1.3.1/test/addrprg/2_fixup_orphan.sam\n","samtools-1.3.1/test/addrprg/2_fixup_orphan.sam.expected\n","samtools-1.3.1/test/addrprg/2_fixup_orphan.sam.expected.err\n","samtools-1.3.1/test/addrprg/3_fixup.sam.expected\n","samtools-1.3.1/test/addrprg/3_fixup.sam.expected.err\n","samtools-1.3.1/test/addrprg/4_fixup_norg.sam\n","samtools-1.3.1/test/addrprg/4_fixup_norg.sam.expected\n","samtools-1.3.1/test/addrprg/4_fixup_norg.sam.expected.err\n","samtools-1.3.1/misc/ace2sam.c\n","samtools-1.3.1/misc/blast2sam.pl\n","samtools-1.3.1/misc/bowtie2sam.pl\n","samtools-1.3.1/misc/export2sam.pl\n","samtools-1.3.1/misc/HmmGlocal.java\n","samtools-1.3.1/misc/interpolate_sam.pl\n","samtools-1.3.1/misc/maq2sam.c\n","samtools-1.3.1/misc/md5fa.c\n","samtools-1.3.1/misc/md5sum-lite.c\n","samtools-1.3.1/misc/novo2sam.pl\n","samtools-1.3.1/misc/plot-bamstats\n","samtools-1.3.1/misc/psl2sam.pl\n","samtools-1.3.1/misc/r2plot.lua\n","samtools-1.3.1/misc/sam2vcf.pl\n","samtools-1.3.1/misc/samtools.pl\n","samtools-1.3.1/misc/seq_cache_populate.pl\n","samtools-1.3.1/misc/soap2sam.pl\n","samtools-1.3.1/misc/varfilter.py\n","samtools-1.3.1/misc/vcfutils.lua\n","samtools-1.3.1/misc/wgsim.1\n","samtools-1.3.1/misc/wgsim.c\n","samtools-1.3.1/misc/wgsim_eval.pl\n","samtools-1.3.1/misc/zoom2sam.pl\n","samtools-1.3.1/htslib-1.3.1/bgzf.c\n","samtools-1.3.1/htslib-1.3.1/bgzip.c\n","samtools-1.3.1/htslib-1.3.1/config.h.in\n","samtools-1.3.1/htslib-1.3.1/config.mk.in\n","samtools-1.3.1/htslib-1.3.1/configure\n","samtools-1.3.1/htslib-1.3.1/configure.ac\n","samtools-1.3.1/htslib-1.3.1/cram/\n","samtools-1.3.1/htslib-1.3.1/faidx.5\n","samtools-1.3.1/htslib-1.3.1/faidx.c\n","samtools-1.3.1/htslib-1.3.1/hfile.c\n","samtools-1.3.1/htslib-1.3.1/hfile_internal.h\n","samtools-1.3.1/htslib-1.3.1/hfile_irods.c\n","samtools-1.3.1/htslib-1.3.1/hfile_libcurl.c\n","samtools-1.3.1/htslib-1.3.1/hfile_net.c\n","samtools-1.3.1/htslib-1.3.1/hts.c\n","samtools-1.3.1/htslib-1.3.1/hts_internal.h\n","samtools-1.3.1/htslib-1.3.1/htsfile.1\n","samtools-1.3.1/htslib-1.3.1/htsfile.c\n","samtools-1.3.1/htslib-1.3.1/htslib/\n","samtools-1.3.1/htslib-1.3.1/htslib.mk\n","samtools-1.3.1/htslib-1.3.1/htslib.pc.in\n","samtools-1.3.1/htslib-1.3.1/htslib_vars.mk\n","samtools-1.3.1/htslib-1.3.1/INSTALL\n","samtools-1.3.1/htslib-1.3.1/kfunc.c\n","samtools-1.3.1/htslib-1.3.1/knetfile.c\n","samtools-1.3.1/htslib-1.3.1/kstring.c\n","samtools-1.3.1/htslib-1.3.1/LICENSE\n","samtools-1.3.1/htslib-1.3.1/Makefile\n","samtools-1.3.1/htslib-1.3.1/md5.c\n","samtools-1.3.1/htslib-1.3.1/NEWS\n","samtools-1.3.1/htslib-1.3.1/plugin.c\n","samtools-1.3.1/htslib-1.3.1/README\n","samtools-1.3.1/htslib-1.3.1/regidx.c\n","samtools-1.3.1/htslib-1.3.1/sam.5\n","samtools-1.3.1/htslib-1.3.1/sam.c\n","samtools-1.3.1/htslib-1.3.1/synced_bcf_reader.c\n","samtools-1.3.1/htslib-1.3.1/tabix.1\n","samtools-1.3.1/htslib-1.3.1/tabix.c\n","samtools-1.3.1/htslib-1.3.1/tbx.c\n","samtools-1.3.1/htslib-1.3.1/test/\n","samtools-1.3.1/htslib-1.3.1/vcf.5\n","samtools-1.3.1/htslib-1.3.1/vcf.c\n","samtools-1.3.1/htslib-1.3.1/vcf_sweep.c\n","samtools-1.3.1/htslib-1.3.1/vcfutils.c\n","samtools-1.3.1/htslib-1.3.1/test/aux#aux_java.cram\n","samtools-1.3.1/htslib-1.3.1/test/auxf#values.sam\n","samtools-1.3.1/htslib-1.3.1/test/auxf.fa\n","samtools-1.3.1/htslib-1.3.1/test/auxf.fa.fai\n","samtools-1.3.1/htslib-1.3.1/test/c1#bounds.sam\n","samtools-1.3.1/htslib-1.3.1/test/c1#clip.sam\n","samtools-1.3.1/htslib-1.3.1/test/c1#noseq.sam\n","samtools-1.3.1/htslib-1.3.1/test/c1#pad1.sam\n","samtools-1.3.1/htslib-1.3.1/test/c1#pad2.sam\n","samtools-1.3.1/htslib-1.3.1/test/c1#pad3.sam\n","samtools-1.3.1/htslib-1.3.1/test/c1#unknown.sam\n","samtools-1.3.1/htslib-1.3.1/test/c1.fa\n","samtools-1.3.1/htslib-1.3.1/test/c1.fa.fai\n","samtools-1.3.1/htslib-1.3.1/test/c2#pad.sam\n","samtools-1.3.1/htslib-1.3.1/test/c2.fa\n","samtools-1.3.1/htslib-1.3.1/test/c2.fa.fai\n","samtools-1.3.1/htslib-1.3.1/test/ce#1.sam\n","samtools-1.3.1/htslib-1.3.1/test/ce#2.sam\n","samtools-1.3.1/htslib-1.3.1/test/ce#5.sam\n","samtools-1.3.1/htslib-1.3.1/test/ce#5b.sam\n","samtools-1.3.1/htslib-1.3.1/test/ce#5b_java.cram\n","samtools-1.3.1/htslib-1.3.1/test/ce#large_seq.sam\n","samtools-1.3.1/htslib-1.3.1/test/ce#supp.sam\n","samtools-1.3.1/htslib-1.3.1/test/ce#tag_depadded.sam\n","samtools-1.3.1/htslib-1.3.1/test/ce#tag_padded.sam\n","samtools-1.3.1/htslib-1.3.1/test/ce#unmap.sam\n","samtools-1.3.1/htslib-1.3.1/test/ce#unmap1.sam\n","samtools-1.3.1/htslib-1.3.1/test/ce#unmap2.sam\n","samtools-1.3.1/htslib-1.3.1/test/ce.fa\n","samtools-1.3.1/htslib-1.3.1/test/ce.fa.fai\n","samtools-1.3.1/htslib-1.3.1/test/compare_sam.pl\n","samtools-1.3.1/htslib-1.3.1/test/cross_validate.sh\n","samtools-1.3.1/htslib-1.3.1/test/faidx.fa\n","samtools-1.3.1/htslib-1.3.1/test/fieldarith.c\n","samtools-1.3.1/htslib-1.3.1/test/fieldarith.sam\n","samtools-1.3.1/htslib-1.3.1/test/formatcols.vcf\n","samtools-1.3.1/htslib-1.3.1/test/hfile.c\n","samtools-1.3.1/htslib-1.3.1/test/noroundtrip-out.vcf\n","samtools-1.3.1/htslib-1.3.1/test/noroundtrip.vcf\n","samtools-1.3.1/htslib-1.3.1/test/sam.c\n","samtools-1.3.1/htslib-1.3.1/test/test-regidx.c\n","samtools-1.3.1/htslib-1.3.1/test/test-vcf-api.c\n","samtools-1.3.1/htslib-1.3.1/test/test-vcf-api.out\n","samtools-1.3.1/htslib-1.3.1/test/test-vcf-hdr-in.vcf\n","samtools-1.3.1/htslib-1.3.1/test/test-vcf-hdr.out\n","samtools-1.3.1/htslib-1.3.1/test/test-vcf-sweep.c\n","samtools-1.3.1/htslib-1.3.1/test/test-vcf-sweep.out\n","samtools-1.3.1/htslib-1.3.1/test/test.pl\n","samtools-1.3.1/htslib-1.3.1/test/test_view.c\n","samtools-1.3.1/htslib-1.3.1/test/xx#blank.sam\n","samtools-1.3.1/htslib-1.3.1/test/xx#large_aux.sam\n","samtools-1.3.1/htslib-1.3.1/test/xx#large_aux2.sam\n","samtools-1.3.1/htslib-1.3.1/test/xx#large_aux_java.cram\n","samtools-1.3.1/htslib-1.3.1/test/xx#minimal.sam\n","samtools-1.3.1/htslib-1.3.1/test/xx#pair.sam\n","samtools-1.3.1/htslib-1.3.1/test/xx#repeated.sam\n","samtools-1.3.1/htslib-1.3.1/test/xx#rg.sam\n","samtools-1.3.1/htslib-1.3.1/test/xx#tlen.sam\n","samtools-1.3.1/htslib-1.3.1/test/xx#tlen2.sam\n","samtools-1.3.1/htslib-1.3.1/test/xx#triplet.sam\n","samtools-1.3.1/htslib-1.3.1/test/xx#unsorted.sam\n","samtools-1.3.1/htslib-1.3.1/test/xx.fa\n","samtools-1.3.1/htslib-1.3.1/test/xx.fa.fai\n","samtools-1.3.1/htslib-1.3.1/htslib/bgzf.h\n","samtools-1.3.1/htslib-1.3.1/htslib/cram.h\n","samtools-1.3.1/htslib-1.3.1/htslib/faidx.h\n","samtools-1.3.1/htslib-1.3.1/htslib/hfile.h\n","samtools-1.3.1/htslib-1.3.1/htslib/hts.h\n","samtools-1.3.1/htslib-1.3.1/htslib/hts_defs.h\n","samtools-1.3.1/htslib-1.3.1/htslib/kbitset.h\n","samtools-1.3.1/htslib-1.3.1/htslib/kfunc.h\n","samtools-1.3.1/htslib-1.3.1/htslib/khash.h\n","samtools-1.3.1/htslib-1.3.1/htslib/khash_str2int.h\n","samtools-1.3.1/htslib-1.3.1/htslib/klist.h\n","samtools-1.3.1/htslib-1.3.1/htslib/knetfile.h\n","samtools-1.3.1/htslib-1.3.1/htslib/kseq.h\n","samtools-1.3.1/htslib-1.3.1/htslib/ksort.h\n","samtools-1.3.1/htslib-1.3.1/htslib/kstring.h\n","samtools-1.3.1/htslib-1.3.1/htslib/regidx.h\n","samtools-1.3.1/htslib-1.3.1/htslib/sam.h\n","samtools-1.3.1/htslib-1.3.1/htslib/synced_bcf_reader.h\n","samtools-1.3.1/htslib-1.3.1/htslib/tbx.h\n","samtools-1.3.1/htslib-1.3.1/htslib/vcf.h\n","samtools-1.3.1/htslib-1.3.1/htslib/vcf_sweep.h\n","samtools-1.3.1/htslib-1.3.1/htslib/vcfutils.h\n","samtools-1.3.1/htslib-1.3.1/cram/cram.h\n","samtools-1.3.1/htslib-1.3.1/cram/cram_codecs.c\n","samtools-1.3.1/htslib-1.3.1/cram/cram_codecs.h\n","samtools-1.3.1/htslib-1.3.1/cram/cram_decode.c\n","samtools-1.3.1/htslib-1.3.1/cram/cram_decode.h\n","samtools-1.3.1/htslib-1.3.1/cram/cram_encode.c\n","samtools-1.3.1/htslib-1.3.1/cram/cram_encode.h\n","samtools-1.3.1/htslib-1.3.1/cram/cram_external.c\n","samtools-1.3.1/htslib-1.3.1/cram/cram_index.c\n","samtools-1.3.1/htslib-1.3.1/cram/cram_index.h\n","samtools-1.3.1/htslib-1.3.1/cram/cram_io.c\n","samtools-1.3.1/htslib-1.3.1/cram/cram_io.h\n","samtools-1.3.1/htslib-1.3.1/cram/cram_samtools.c\n","samtools-1.3.1/htslib-1.3.1/cram/cram_samtools.h\n","samtools-1.3.1/htslib-1.3.1/cram/cram_stats.c\n","samtools-1.3.1/htslib-1.3.1/cram/cram_stats.h\n","samtools-1.3.1/htslib-1.3.1/cram/cram_structs.h\n","samtools-1.3.1/htslib-1.3.1/cram/files.c\n","samtools-1.3.1/htslib-1.3.1/cram/mFILE.c\n","samtools-1.3.1/htslib-1.3.1/cram/mFILE.h\n","samtools-1.3.1/htslib-1.3.1/cram/misc.h\n","samtools-1.3.1/htslib-1.3.1/cram/open_trace_file.c\n","samtools-1.3.1/htslib-1.3.1/cram/open_trace_file.h\n","samtools-1.3.1/htslib-1.3.1/cram/os.h\n","samtools-1.3.1/htslib-1.3.1/cram/pooled_alloc.c\n","samtools-1.3.1/htslib-1.3.1/cram/pooled_alloc.h\n","samtools-1.3.1/htslib-1.3.1/cram/rANS_byte.h\n","samtools-1.3.1/htslib-1.3.1/cram/rANS_static.c\n","samtools-1.3.1/htslib-1.3.1/cram/rANS_static.h\n","samtools-1.3.1/htslib-1.3.1/cram/sam_header.c\n","samtools-1.3.1/htslib-1.3.1/cram/sam_header.h\n","samtools-1.3.1/htslib-1.3.1/cram/string_alloc.c\n","samtools-1.3.1/htslib-1.3.1/cram/string_alloc.h\n","samtools-1.3.1/htslib-1.3.1/cram/thread_pool.c\n","samtools-1.3.1/htslib-1.3.1/cram/thread_pool.h\n","samtools-1.3.1/htslib-1.3.1/cram/vlen.c\n","samtools-1.3.1/htslib-1.3.1/cram/vlen.h\n","samtools-1.3.1/htslib-1.3.1/cram/zfio.c\n","samtools-1.3.1/htslib-1.3.1/cram/zfio.h\n","samtools-1.3.1/examples/00README.txt\n","samtools-1.3.1/examples/ex1.fa\n","samtools-1.3.1/examples/ex1.sam.gz\n","samtools-1.3.1/examples/toy.fa\n","samtools-1.3.1/examples/toy.sam\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   utils.c -o utils.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   kthread.c -o kthread.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   kstring.c -o kstring.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   ksw.c -o ksw.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bwt.c -o bwt.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bntseq.c -o bntseq.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bwa.c -o bwa.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bwamem.c -o bwamem.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bwamem_pair.c -o bwamem_pair.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bwamem_extra.c -o bwamem_extra.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   malloc_wrap.c -o malloc_wrap.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   QSufSort.c -o QSufSort.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bwt_gen.c -o bwt_gen.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   rope.c -o rope.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   rle.c -o rle.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   is.c -o is.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bwtindex.c -o bwtindex.o\n","ar -csru libbwa.a utils.o kthread.o kstring.o ksw.o bwt.o bntseq.o bwa.o bwamem.o bwamem_pair.o bwamem_extra.o malloc_wrap.o QSufSort.o bwt_gen.o rope.o rle.o is.o bwtindex.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bwashm.c -o bwashm.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bwase.c -o bwase.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bwaseqio.c -o bwaseqio.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bwtgap.c -o bwtgap.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bwtaln.c -o bwtaln.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bamlite.c -o bamlite.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bwape.c -o bwape.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   kopen.c -o kopen.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   pemerge.c -o pemerge.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   maxk.c -o maxk.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bwtsw2_core.c -o bwtsw2_core.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bwtsw2_main.c -o bwtsw2_main.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bwtsw2_aux.c -o bwtsw2_aux.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bwt_lite.c -o bwt_lite.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bwtsw2_chain.c -o bwtsw2_chain.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   fastmap.c -o fastmap.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   bwtsw2_pair.c -o bwtsw2_pair.o\n","gcc -c -g -Wall -Wno-unused-function -O2 -DHAVE_PTHREAD -DUSE_MALLOC_WRAPPERS   main.c -o main.o\n","gcc -g -Wall -Wno-unused-function -O2  bwashm.o bwase.o bwaseqio.o bwtgap.o bwtaln.o bamlite.o bwape.o kopen.o pemerge.o maxk.o bwtsw2_core.o bwtsw2_main.o bwtsw2_aux.o bwt_lite.o bwtsw2_chain.o fastmap.o bwtsw2_pair.o main.o -o bwa -L. -lbwa -lm -lz -lpthread -lrt\n","echo '/* Basic config.h generated by Makefile */' > config.h\n","echo '#define HAVE_CURSES' >> config.h\n","echo '#define HAVE_CURSES_H' >> config.h\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_index.o bam_index.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_plcmd.o bam_plcmd.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o sam_view.o sam_view.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_cat.o bam_cat.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_md.o bam_md.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_reheader.o bam_reheader.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_sort.o bam_sort.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bedidx.o bedidx.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o kprobaln.o kprobaln.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_rmdup.o bam_rmdup.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_rmdupse.o bam_rmdupse.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_mate.o bam_mate.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_stat.o bam_stat.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_color.o bam_color.c\n","echo '#define SAMTOOLS_VERSION \"1.3.1\"' > version.h\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bamtk.o bamtk.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam2bcf.o bam2bcf.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam2bcf_indel.o bam2bcf_indel.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o errmod.o errmod.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o sample.o sample.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o sam_opts.o sam_opts.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o cut_target.o cut_target.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o phase.o phase.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam2depth.o bam2depth.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o padding.o padding.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bedcov.o bedcov.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bamshuf.o bamshuf.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o faidx.o faidx.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o dict.o dict.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o stats.o stats.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o stats_isize.o stats_isize.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_flags.o bam_flags.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_split.o bam_split.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_tview.o bam_tview.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_tview_curses.o bam_tview_curses.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_tview_html.o bam_tview_html.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_lpileup.o bam_lpileup.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_quickcheck.o bam_quickcheck.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_addrprg.o bam_addrprg.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_aux.o bam_aux.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam.o bam.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_import.o bam_import.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o sam.o sam.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o sam_header.o sam_header.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o bam_plbuf.o bam_plbuf.c\n","ar -csru libbam.a bam_aux.o bam.o bam_import.o sam.o sam_header.o bam_plbuf.o\n","cd htslib-1.3.1 && make config.h\n","make[1]: Entering directory '/content/software/samtools/htslib-1.3.1'\n","echo '/* Empty config.h generated by Makefile */' > config.h\n","make[1]: Leaving directory '/content/software/samtools/htslib-1.3.1'\n","cd htslib-1.3.1 && make lib-static\n","make[1]: Entering directory '/content/software/samtools/htslib-1.3.1'\n","gcc -g -Wall -O2 -I.  -c -o kfunc.o kfunc.c\n","gcc -g -Wall -O2 -I.  -c -o knetfile.o knetfile.c\n","gcc -g -Wall -O2 -I.  -c -o kstring.o kstring.c\n","gcc -g -Wall -O2 -I.  -c -o bgzf.o bgzf.c\n","gcc -g -Wall -O2 -I.  -c -o faidx.o faidx.c\n","gcc -g -Wall -O2 -I.  -c -o hfile.o hfile.c\n","gcc -g -Wall -O2 -I.  -c -o hfile_net.o hfile_net.c\n","echo '#define HTS_VERSION \"1.3.1\"' > version.h\n","gcc -g -Wall -O2 -I.  -c -o hts.o hts.c\n","gcc -g -Wall -O2 -I.  -c -o md5.o md5.c\n","gcc -g -Wall -O2 -I.  -c -o regidx.o regidx.c\n","gcc -g -Wall -O2 -I.  -c -o sam.o sam.c\n","gcc -g -Wall -O2 -I.  -c -o synced_bcf_reader.o synced_bcf_reader.c\n","gcc -g -Wall -O2 -I.  -c -o vcf_sweep.o vcf_sweep.c\n","gcc -g -Wall -O2 -I.  -c -o tbx.o tbx.c\n","gcc -g -Wall -O2 -I.  -c -o vcf.o vcf.c\n","gcc -g -Wall -O2 -I.  -c -o vcfutils.o vcfutils.c\n","gcc -g -Wall -O2 -I.  -c -o cram/cram_codecs.o cram/cram_codecs.c\n","gcc -g -Wall -O2 -I.  -c -o cram/cram_decode.o cram/cram_decode.c\n","gcc -g -Wall -O2 -I.  -c -o cram/cram_encode.o cram/cram_encode.c\n","gcc -g -Wall -O2 -I.  -c -o cram/cram_external.o cram/cram_external.c\n","gcc -g -Wall -O2 -I.  -c -o cram/cram_index.o cram/cram_index.c\n","gcc -g -Wall -O2 -I.  -c -o cram/cram_io.o cram/cram_io.c\n","gcc -g -Wall -O2 -I.  -c -o cram/cram_samtools.o cram/cram_samtools.c\n","gcc -g -Wall -O2 -I.  -c -o cram/cram_stats.o cram/cram_stats.c\n","gcc -g -Wall -O2 -I.  -c -o cram/files.o cram/files.c\n","gcc -g -Wall -O2 -I.  -c -o cram/mFILE.o cram/mFILE.c\n","gcc -g -Wall -O2 -I.  -c -o cram/open_trace_file.o cram/open_trace_file.c\n","gcc -g -Wall -O2 -I.  -c -o cram/pooled_alloc.o cram/pooled_alloc.c\n","gcc -g -Wall -O2 -I.  -c -o cram/rANS_static.o cram/rANS_static.c\n","gcc -g -Wall -O2 -I.  -c -o cram/sam_header.o cram/sam_header.c\n","gcc -g -Wall -O2 -I.  -c -o cram/string_alloc.o cram/string_alloc.c\n","gcc -g -Wall -O2 -I.  -c -o cram/thread_pool.o cram/thread_pool.c\n","gcc -g -Wall -O2 -I.  -c -o cram/vlen.o cram/vlen.c\n","gcc -g -Wall -O2 -I.  -c -o cram/zfio.o cram/zfio.c\n","ar -rc libhts.a kfunc.o knetfile.o kstring.o bgzf.o faidx.o hfile.o hfile_net.o hts.o md5.o regidx.o sam.o synced_bcf_reader.o vcf_sweep.o tbx.o vcf.o vcfutils.o cram/cram_codecs.o cram/cram_decode.o cram/cram_encode.o cram/cram_external.o cram/cram_index.o cram/cram_io.o cram/cram_samtools.o cram/cram_stats.o cram/files.o cram/mFILE.o cram/open_trace_file.o cram/pooled_alloc.o cram/rANS_static.o cram/sam_header.o cram/string_alloc.o cram/thread_pool.o cram/vlen.o cram/zfio.o\n","ranlib libhts.a\n","make[1]: Leaving directory '/content/software/samtools/htslib-1.3.1'\n","gcc -pthread -rdynamic   -o samtools bam_index.o bam_plcmd.o sam_view.o bam_cat.o bam_md.o bam_reheader.o bam_sort.o bedidx.o kprobaln.o bam_rmdup.o bam_rmdupse.o bam_mate.o bam_stat.o bam_color.o bamtk.o bam2bcf.o bam2bcf_indel.o errmod.o sample.o sam_opts.o cut_target.o phase.o bam2depth.o padding.o bedcov.o bamshuf.o faidx.o dict.o stats.o stats_isize.o bam_flags.o bam_split.o bam_tview.o bam_tview_curses.o bam_tview_html.o bam_lpileup.o bam_quickcheck.o bam_addrprg.o libbam.a htslib-1.3.1/libhts.a -lcurses -lm -lz -ldl \n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o misc/ace2sam.o misc/ace2sam.c\n","gcc  -o misc/ace2sam misc/ace2sam.o -lz -ldl \n","gcc -g -Wall -O2 -DMAQ_LONGREADS -I. -Ihtslib-1.3.1  -c -o misc/maq2sam-long.o misc/maq2sam.c\n","gcc  -o misc/maq2sam-long misc/maq2sam-long.o -lz -ldl \n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o misc/maq2sam-short.o misc/maq2sam.c\n","gcc  -o misc/maq2sam-short misc/maq2sam-short.o -lz -ldl \n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o misc/md5fa.o misc/md5fa.c\n","gcc -rdynamic   -o misc/md5fa misc/md5fa.o htslib-1.3.1/libhts.a -lz -ldl \n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o misc/md5sum-lite.o misc/md5sum-lite.c\n","gcc -rdynamic   -o misc/md5sum-lite misc/md5sum-lite.o htslib-1.3.1/libhts.a -lz -ldl \n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o misc/wgsim.o misc/wgsim.c\n","gcc  -o misc/wgsim misc/wgsim.o -lm -lz -ldl \n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o test/merge/test_bam_translate.o test/merge/test_bam_translate.c\n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o test/test.o test/test.c\n","gcc -pthread -rdynamic   -o test/merge/test_bam_translate test/merge/test_bam_translate.o test/test.o sam_opts.o htslib-1.3.1/libhts.a -lz -ldl \n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o test/merge/test_rtrans_build.o test/merge/test_rtrans_build.c\n","gcc -pthread -rdynamic   -o test/merge/test_rtrans_build test/merge/test_rtrans_build.o sam_opts.o htslib-1.3.1/libhts.a -lz -ldl \n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o test/merge/test_trans_tbl_init.o test/merge/test_trans_tbl_init.c\n","gcc -pthread -rdynamic   -o test/merge/test_trans_tbl_init test/merge/test_trans_tbl_init.o sam_opts.o htslib-1.3.1/libhts.a -lz -ldl \n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o test/split/test_count_rg.o test/split/test_count_rg.c\n","gcc -pthread -rdynamic   -o test/split/test_count_rg test/split/test_count_rg.o test/test.o sam_opts.o htslib-1.3.1/libhts.a -lz -ldl \n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o test/split/test_expand_format_string.o test/split/test_expand_format_string.c\n","gcc -pthread -rdynamic   -o test/split/test_expand_format_string test/split/test_expand_format_string.o test/test.o sam_opts.o htslib-1.3.1/libhts.a -lz -ldl \n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o test/split/test_filter_header_rg.o test/split/test_filter_header_rg.c\n","gcc -pthread -rdynamic   -o test/split/test_filter_header_rg test/split/test_filter_header_rg.o test/test.o sam_opts.o htslib-1.3.1/libhts.a -lz -ldl \n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o test/split/test_parse_args.o test/split/test_parse_args.c\n","gcc -pthread -rdynamic   -o test/split/test_parse_args test/split/test_parse_args.o test/test.o sam_opts.o htslib-1.3.1/libhts.a -lz -ldl \n","gcc -g -Wall -O2 -I. -Ihtslib-1.3.1  -c -o test/vcf-miniview.o test/vcf-miniview.c\n","gcc -pthread -rdynamic   -o test/vcf-miniview test/vcf-miniview.o htslib-1.3.1/libhts.a -lz -ldl \n","mkdir -p -m 755 /usr/local/bin /usr/local/share/man/man1\n","install -p samtools misc/ace2sam misc/maq2sam-long misc/maq2sam-short misc/md5fa misc/md5sum-lite misc/wgsim misc/blast2sam.pl misc/bowtie2sam.pl misc/export2sam.pl misc/interpolate_sam.pl misc/novo2sam.pl misc/plot-bamstats misc/psl2sam.pl misc/sam2vcf.pl misc/samtools.pl misc/seq_cache_populate.pl misc/soap2sam.pl misc/varfilter.py misc/wgsim_eval.pl misc/zoom2sam.pl /usr/local/bin\n","install -p -m 644 samtools.1 misc/wgsim.1 /usr/local/share/man/man1\n"]},{"output_type":"stream","name":"stderr","text":["Cloning into 'bwa'...\n","--2022-06-29 16:45:39--  https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/samtools-1.3.1.tar.bz2\n","Resolving storage.googleapis.com (storage.googleapis.com)... 74.125.204.128, 64.233.189.128, 74.125.23.128, ...\n","Connecting to storage.googleapis.com (storage.googleapis.com)|74.125.204.128|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 4030072 (3.8M) [application/octet-stream]\n","Saving to: ‘samtools.tar.bz2’\n","\n","     0K .......... .......... .......... .......... ..........  1% 20.5M 0s\n","    50K .......... .......... .......... .......... ..........  2% 30.5M 0s\n","   100K .......... .......... .......... .......... ..........  3% 7.18M 0s\n","   150K .......... .......... .......... .......... ..........  5% 33.2M 0s\n","   200K .......... .......... .......... .......... ..........  6% 5.98M 0s\n","   250K .......... .......... .......... .......... ..........  7% 29.3M 0s\n","   300K .......... .......... .......... .......... ..........  8% 33.7M 0s\n","   350K .......... .......... .......... .......... .......... 10% 10.1M 0s\n","   400K .......... .......... .......... .......... .......... 11% 21.8M 0s\n","   450K .......... .......... .......... .......... .......... 12% 21.7M 0s\n","   500K .......... .......... .......... .......... .......... 13% 21.7M 0s\n","   550K .......... .......... .......... .......... .......... 15% 14.0M 0s\n","   600K .......... .......... .......... .......... .......... 16% 16.0M 0s\n","   650K .......... .......... .......... .......... .......... 17% 17.9M 0s\n","   700K .......... .......... .......... .......... .......... 19% 23.1M 0s\n","   750K .......... .......... .......... .......... .......... 20% 25.0M 0s\n","   800K .......... .......... .......... .......... .......... 21% 20.0M 0s\n","   850K .......... .......... .......... .......... .......... 22% 24.5M 0s\n","   900K .......... .......... .......... .......... .......... 24% 23.3M 0s\n","   950K .......... .......... .......... .......... .......... 25% 24.6M 0s\n","  1000K .......... .......... .......... .......... .......... 26% 23.9M 0s\n","  1050K .......... .......... .......... .......... .......... 27% 28.4M 0s\n","  1100K .......... .......... .......... .......... .......... 29% 33.0M 0s\n","  1150K .......... .......... .......... .......... .......... 30% 36.2M 0s\n","  1200K .......... .......... .......... .......... .......... 31% 27.8M 0s\n","  1250K .......... .......... .......... .......... .......... 33% 23.3M 0s\n","  1300K .......... .......... .......... .......... .......... 34% 22.5M 0s\n","  1350K .......... .......... .......... .......... .......... 35% 27.0M 0s\n","  1400K .......... .......... .......... .......... .......... 36% 15.8M 0s\n","  1450K .......... .......... .......... .......... .......... 38% 27.0M 0s\n","  1500K .......... .......... .......... .......... .......... 39% 40.6M 0s\n","  1550K .......... .......... .......... .......... .......... 40%  201M 0s\n","  1600K .......... .......... .......... .......... .......... 41%  157M 0s\n","  1650K .......... .......... .......... .......... .......... 43%  114M 0s\n","  1700K .......... .......... .......... .......... .......... 44%  116M 0s\n","  1750K .......... .......... .......... .......... .......... 45%  108M 0s\n","  1800K .......... .......... .......... .......... .......... 47%  167M 0s\n","  1850K .......... .......... .......... .......... .......... 48%  202M 0s\n","  1900K .......... .......... .......... .......... .......... 49%  179M 0s\n","  1950K .......... .......... .......... .......... .......... 50%  195M 0s\n","  2000K .......... .......... .......... .......... .......... 52%  173M 0s\n","  2050K .......... .......... .......... .......... .......... 53% 41.6M 0s\n","  2100K .......... .......... .......... .......... .......... 54% 33.8M 0s\n","  2150K .......... .......... .......... .......... .......... 55% 29.5M 0s\n","  2200K .......... .......... .......... .......... .......... 57% 28.2M 0s\n","  2250K .......... .......... .......... .......... .......... 58% 35.4M 0s\n","  2300K .......... .......... .......... .......... .......... 59% 34.7M 0s\n","  2350K .......... .......... .......... .......... .......... 60% 36.2M 0s\n","  2400K .......... .......... .......... .......... .......... 62% 24.0M 0s\n","  2450K .......... .......... .......... .......... .......... 63% 49.5M 0s\n","  2500K .......... .......... .......... .......... .......... 64% 51.2M 0s\n","  2550K .......... .......... .......... .......... .......... 66% 10.5M 0s\n","  2600K .......... .......... .......... .......... .......... 67% 25.7M 0s\n","  2650K .......... .......... .......... .......... .......... 68% 31.8M 0s\n","  2700K .......... .......... .......... .......... .......... 69% 22.4M 0s\n","  2750K .......... .......... .......... .......... .......... 71%  174M 0s\n","  2800K .......... .......... .......... .......... .......... 72% 3.60M 0s\n","  2850K .......... .......... .......... .......... .......... 73% 47.8M 0s\n","  2900K .......... .......... .......... .......... .......... 74%  171M 0s\n","  2950K .......... .......... .......... .......... .......... 76%  203M 0s\n","  3000K .......... .......... .......... .......... .......... 77%  168M 0s\n","  3050K .......... .......... .......... .......... .......... 78% 6.49M 0s\n","  3100K .......... .......... .......... .......... .......... 80% 39.5M 0s\n","  3150K .......... .......... .......... .......... .......... 81% 26.2M 0s\n","  3200K .......... .......... .......... .......... .......... 82% 16.7M 0s\n","  3250K .......... .......... .......... .......... .......... 83%  165M 0s\n","  3300K .......... .......... .......... .......... .......... 85% 3.77M 0s\n","  3350K .......... .......... .......... .......... .......... 86% 44.5M 0s\n","  3400K .......... .......... .......... .......... .......... 87%  195M 0s\n","  3450K .......... .......... .......... .......... .......... 88%  249M 0s\n","  3500K .......... .......... .......... .......... .......... 90%  223M 0s\n","  3550K .......... .......... .......... .......... .......... 91% 3.94M 0s\n","  3600K .......... .......... .......... .......... .......... 92% 15.4M 0s\n","  3650K .......... .......... .......... .......... .......... 94% 45.6M 0s\n","  3700K .......... .......... .......... .......... .......... 95% 93.9M 0s\n","  3750K .......... .......... .......... .......... .......... 96%  225M 0s\n","  3800K .......... .......... .......... .......... .......... 97% 24.4M 0s\n","  3850K .......... .......... .......... .......... .......... 99% 17.4M 0s\n","  3900K .......... .......... .......... .....                100% 11.3M=0.2s\n","\n","2022-06-29 16:45:40 (22.0 MB/s) - ‘samtools.tar.bz2’ saved [4030072/4030072]\n","\n","--2022-06-29 16:45:41--  https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/SPAdes-3.12.0-Linux.tar.gz\n","Resolving storage.googleapis.com (storage.googleapis.com)... 74.125.204.128, 64.233.189.128, 74.125.23.128, ...\n","Connecting to storage.googleapis.com (storage.googleapis.com)|74.125.204.128|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 17106384 (16M) [application/x-gzip]\n","Saving to: ‘SPAdes-3.12.0-Linux.tar.gz’\n","\n","     0K .......... .......... .......... .......... ..........  0% 49.8M 0s\n","    50K .......... .......... .......... .......... ..........  0% 67.6M 0s\n","   100K .......... .......... .......... .......... ..........  0% 65.2M 0s\n","   150K .......... .......... .......... .......... ..........  1% 35.5M 0s\n","   200K .......... .......... .......... .......... ..........  1% 32.1M 0s\n","   250K .......... .......... .......... .......... ..........  1% 27.0M 0s\n","   300K .......... .......... .......... .......... ..........  2% 31.8M 0s\n","   350K .......... .......... .......... .......... ..........  2% 33.2M 0s\n","   400K .......... .......... .......... .......... ..........  2% 27.1M 0s\n","   450K .......... .......... .......... .......... ..........  2% 32.3M 0s\n","   500K .......... .......... .......... .......... ..........  3% 31.0M 0s\n","   550K .......... .......... .......... .......... ..........  3% 31.5M 0s\n","   600K .......... .......... .......... .......... ..........  3% 25.8M 0s\n","   650K .......... .......... .......... .......... ..........  4% 33.8M 0s\n","   700K .......... .......... .......... .......... ..........  4% 31.9M 0s\n","   750K .......... .......... .......... .......... ..........  4% 34.0M 0s\n","   800K .......... .......... .......... .......... ..........  5% 27.6M 0s\n","   850K .......... .......... .......... .......... ..........  5% 26.6M 0s\n","   900K .......... .......... .......... .......... ..........  5% 31.6M 0s\n","   950K .......... .......... .......... .......... ..........  5% 33.4M 0s\n","  1000K .......... .......... .......... .......... ..........  6% 33.6M 0s\n","  1050K .......... .......... .......... .......... ..........  6% 41.5M 0s\n","  1100K .......... .......... .......... .......... ..........  6% 44.9M 0s\n","  1150K .......... .......... .......... .......... ..........  7% 40.2M 0s\n","  1200K .......... .......... .......... .......... ..........  7% 46.4M 0s\n","  1250K .......... .......... .......... .......... ..........  7% 36.7M 0s\n","  1300K .......... .......... .......... .......... ..........  8% 38.5M 0s\n","  1350K .......... .......... .......... .......... ..........  8% 41.7M 0s\n","  1400K .......... .......... .......... .......... ..........  8% 46.1M 0s\n","  1450K .......... .......... .......... .......... ..........  8% 45.6M 0s\n","  1500K .......... .......... .......... .......... ..........  9%  170M 0s\n","  1550K .......... .......... .......... .......... ..........  9%  130M 0s\n","  1600K .......... .......... .......... .......... ..........  9%  113M 0s\n","  1650K .......... .......... .......... .......... .......... 10%  136M 0s\n","  1700K .......... .......... .......... .......... .......... 10%  133M 0s\n","  1750K .......... .......... .......... .......... .......... 10% 91.4M 0s\n","  1800K .......... .......... .......... .......... .......... 11%  118M 0s\n","  1850K .......... .......... .......... .......... .......... 11%  151M 0s\n","  1900K .......... .......... .......... .......... .......... 11% 37.2M 0s\n","  1950K .......... .......... .......... .......... .......... 11% 39.0M 0s\n","  2000K .......... .......... .......... .......... .......... 12% 47.0M 0s\n","  2050K .......... .......... .......... .......... .......... 12% 40.9M 0s\n","  2100K .......... .......... .......... .......... .......... 12% 38.0M 0s\n","  2150K .......... .......... .......... .......... .......... 13% 38.5M 0s\n","  2200K .......... .......... .......... .......... .......... 13% 46.5M 0s\n","  2250K .......... .......... .......... .......... .......... 13% 48.4M 0s\n","  2300K .......... .......... .......... .......... .......... 14% 39.1M 0s\n","  2350K .......... .......... .......... .......... .......... 14% 39.9M 0s\n","  2400K .......... .......... .......... .......... .......... 14% 48.6M 0s\n","  2450K .......... .......... .......... .......... .......... 14% 65.6M 0s\n","  2500K .......... .......... .......... .......... .......... 15% 79.2M 0s\n","  2550K .......... .......... .......... .......... .......... 15%  187M 0s\n","  2600K .......... .......... .......... .......... .......... 15%  143M 0s\n","  2650K .......... .......... .......... .......... .......... 16% 67.8M 0s\n","  2700K .......... .......... .......... .......... .......... 16% 48.7M 0s\n","  2750K .......... .......... .......... .......... .......... 16% 37.0M 0s\n","  2800K .......... .......... .......... .......... .......... 17% 30.5M 0s\n","  2850K .......... .......... .......... .......... .......... 17%  130M 0s\n","  2900K .......... .......... .......... .......... .......... 17%  194M 0s\n","  2950K .......... .......... .......... .......... .......... 17%  189M 0s\n","  3000K .......... .......... .......... .......... .......... 18%  188M 0s\n","  3050K .......... .......... .......... .......... .......... 18%  167M 0s\n","  3100K .......... .......... .......... .......... .......... 18% 56.7M 0s\n","  3150K .......... .......... .......... .......... .......... 19% 63.6M 0s\n","  3200K .......... .......... .......... .......... .......... 19% 44.5M 0s\n","  3250K .......... .......... .......... .......... .......... 19% 38.6M 0s\n","  3300K .......... .......... .......... .......... .......... 20% 59.8M 0s\n","  3350K .......... .......... .......... .......... .......... 20%  186M 0s\n","  3400K .......... .......... .......... .......... .......... 20%  201M 0s\n","  3450K .......... .......... .......... .......... .......... 20%  201M 0s\n","  3500K .......... .......... .......... .......... .......... 21%  212M 0s\n","  3550K .......... .......... .......... .......... .......... 21%  131M 0s\n","  3600K .......... .......... .......... .......... .......... 21%  159M 0s\n","  3650K .......... .......... .......... .......... .......... 22%  198M 0s\n","  3700K .......... .......... .......... .......... .......... 22%  235M 0s\n","  3750K .......... .......... .......... .......... .......... 22%  141M 0s\n","  3800K .......... .......... .......... .......... .......... 23%  148M 0s\n","  3850K .......... .......... .......... .......... .......... 23%  175M 0s\n","  3900K .......... .......... .......... .......... .......... 23%  186M 0s\n","  3950K .......... .......... .......... .......... .......... 23%  188M 0s\n","  4000K .......... .......... .......... .......... .......... 24%  208M 0s\n","  4050K .......... .......... .......... .......... .......... 24%  138M 0s\n","  4100K .......... .......... .......... .......... .......... 24% 79.5M 0s\n","  4150K .......... .......... .......... .......... .......... 25% 44.7M 0s\n","  4200K .......... .......... .......... .......... .......... 25% 37.8M 0s\n","  4250K .......... .......... .......... .......... .......... 25% 39.9M 0s\n","  4300K .......... .......... .......... .......... .......... 26% 46.6M 0s\n","  4350K .......... .......... .......... .......... .......... 26% 46.9M 0s\n","  4400K .......... .......... .......... .......... .......... 26% 40.6M 0s\n","  4450K .......... .......... .......... .......... .......... 26% 42.7M 0s\n","  4500K .......... .......... .......... .......... .......... 27% 41.2M 0s\n","  4550K .......... .......... .......... .......... .......... 27% 46.5M 0s\n","  4600K .......... .......... .......... .......... .......... 27% 42.6M 0s\n","  4650K .......... .......... .......... .......... .......... 28% 49.4M 0s\n","  4700K .......... .......... .......... .......... .......... 28%  111M 0s\n","  4750K .......... .......... .......... .......... .......... 28%  251M 0s\n","  4800K .......... .......... .......... .......... .......... 29%  159M 0s\n","  4850K .......... .......... .......... .......... .......... 29%  158M 0s\n","  4900K .......... .......... .......... .......... .......... 29%  172M 0s\n","  4950K .......... .......... .......... .......... .......... 29%  151M 0s\n","  5000K .......... .......... .......... .......... .......... 30%  171M 0s\n","  5050K .......... .......... .......... .......... .......... 30%  250M 0s\n","  5100K .......... .......... .......... .......... .......... 30%  259M 0s\n","  5150K .......... .......... .......... .......... .......... 31%  209M 0s\n","  5200K .......... .......... .......... .......... .......... 31%  216M 0s\n","  5250K .......... .......... .......... .......... .......... 31%  142M 0s\n","  5300K .......... .......... .......... .......... .......... 32%  172M 0s\n","  5350K .......... .......... .......... .......... .......... 32%  236M 0s\n","  5400K .......... .......... .......... .......... .......... 32%  197M 0s\n","  5450K .......... .......... .......... .......... .......... 32%  188M 0s\n","  5500K .......... .......... .......... .......... .......... 33%  152M 0s\n","  5550K .......... .......... .......... .......... .......... 33%  200M 0s\n","  5600K .......... .......... .......... .......... .......... 33%  107M 0s\n","  5650K .......... .......... .......... .......... .......... 34% 44.3M 0s\n","  5700K .......... .......... .......... .......... .......... 34% 37.6M 0s\n","  5750K .......... .......... .......... .......... .......... 34% 33.5M 0s\n","  5800K .......... .......... .......... .......... .......... 35% 43.0M 0s\n","  5850K .......... .......... .......... .......... .......... 35% 48.4M 0s\n","  5900K .......... .......... .......... .......... .......... 35% 44.5M 0s\n","  5950K .......... .......... .......... .......... .......... 35% 37.1M 0s\n","  6000K .......... .......... .......... .......... .......... 36% 40.7M 0s\n","  6050K .......... .......... .......... .......... .......... 36% 48.4M 0s\n","  6100K .......... .......... .......... .......... .......... 36% 48.0M 0s\n","  6150K .......... .......... .......... .......... .......... 37% 45.9M 0s\n","  6200K .......... .......... .......... .......... .......... 37% 35.1M 0s\n","  6250K .......... .......... .......... .......... .......... 37% 46.0M 0s\n","  6300K .......... .......... .......... .......... .......... 38%  128M 0s\n","  6350K .......... .......... .......... .......... .......... 38%  209M 0s\n","  6400K .......... .......... .......... .......... .......... 38%  173M 0s\n","  6450K .......... .......... .......... .......... .......... 38%  243M 0s\n","  6500K .......... .......... .......... .......... .......... 39%  143M 0s\n","  6550K .......... .......... .......... .......... .......... 39%  156M 0s\n","  6600K .......... .......... .......... .......... .......... 39%  177M 0s\n","  6650K .......... .......... .......... .......... .......... 40%  217M 0s\n","  6700K .......... .......... .......... .......... .......... 40%  231M 0s\n","  6750K .......... .......... .......... .......... .......... 40%  164M 0s\n","  6800K .......... .......... .......... .......... .......... 41%  183M 0s\n","  6850K .......... .......... .......... .......... .......... 41%  195M 0s\n","  6900K .......... .......... .......... .......... .......... 41%  203M 0s\n","  6950K .......... .......... .......... .......... .......... 41%  147M 0s\n","  7000K .......... .......... .......... .......... .......... 42%  221M 0s\n","  7050K .......... .......... .......... .......... .......... 42%  237M 0s\n","  7100K .......... .......... .......... .......... .......... 42%  226M 0s\n","  7150K .......... .......... .......... .......... .......... 43% 67.5M 0s\n","  7200K .......... .......... .......... .......... .......... 43% 42.7M 0s\n","  7250K .......... .......... .......... .......... .......... 43% 48.4M 0s\n","  7300K .......... .......... .......... .......... .......... 43% 88.5M 0s\n","  7350K .......... .......... .......... .......... .......... 44% 48.2M 0s\n","  7400K .......... .......... .......... .......... .......... 44% 43.1M 0s\n","  7450K .......... .......... .......... .......... .......... 44% 48.4M 0s\n","  7500K .......... .......... .......... .......... .......... 45% 49.1M 0s\n","  7550K .......... .......... .......... .......... .......... 45% 49.8M 0s\n","  7600K .......... .......... .......... .......... .......... 45% 42.7M 0s\n","  7650K .......... .......... .......... .......... .......... 46% 44.5M 0s\n","  7700K .......... .......... .......... .......... .......... 46% 47.2M 0s\n","  7750K .......... .......... .......... .......... .......... 46% 47.9M 0s\n","  7800K .......... .......... .......... .......... .......... 46% 40.7M 0s\n","  7850K .......... .......... .......... .......... .......... 47% 64.8M 0s\n","  7900K .......... .......... .......... .......... .......... 47%  255M 0s\n","  7950K .......... .......... .......... .......... .......... 47%  231M 0s\n","  8000K .......... .......... .......... .......... .......... 48%  221M 0s\n","  8050K .......... .......... .......... .......... .......... 48%  239M 0s\n","  8100K .......... .......... .......... .......... .......... 48%  252M 0s\n","  8150K .......... .......... .......... .......... .......... 49%  347K 0s\n","  8200K .......... .......... .......... .......... .......... 49%  180M 0s\n","  8250K .......... .......... .......... .......... .......... 49%  210M 0s\n","  8300K .......... .......... .......... .......... .......... 49%  268M 0s\n","  8350K .......... .......... .......... .......... .......... 50%  307M 0s\n","  8400K .......... .......... .......... .......... .......... 50%  231M 0s\n","  8450K .......... .......... .......... .......... .......... 50%  377M 0s\n","  8500K .......... .......... .......... .......... .......... 51%  196M 0s\n","  8550K .......... .......... .......... .......... .......... 51%  253M 0s\n","  8600K .......... .......... .......... .......... .......... 51%  179M 0s\n","  8650K .......... .......... .......... .......... .......... 52%  207M 0s\n","  8700K .......... .......... .......... .......... .......... 52%  196M 0s\n","  8750K .......... .......... .......... .......... .......... 52%  259M 0s\n","  8800K .......... .......... .......... .......... .......... 52%  154M 0s\n","  8850K .......... .......... .......... .......... .......... 53%  228M 0s\n","  8900K .......... .......... .......... .......... .......... 53% 37.8M 0s\n","  8950K .......... .......... .......... .......... .......... 53% 41.8M 0s\n","  9000K .......... .......... .......... .......... .......... 54% 34.8M 0s\n","  9050K .......... .......... .......... .......... .......... 54% 41.9M 0s\n","  9100K .......... .......... .......... .......... .......... 54% 23.8M 0s\n","  9150K .......... .......... .......... .......... .......... 55%  125M 0s\n","  9200K .......... .......... .......... .......... .......... 55% 43.1M 0s\n","  9250K .......... .......... .......... .......... .......... 55% 34.2M 0s\n","  9300K .......... .......... .......... .......... .......... 55% 31.0M 0s\n","  9350K .......... .......... .......... .......... .......... 56% 26.3M 0s\n","  9400K .......... .......... .......... .......... .......... 56% 73.5M 0s\n","  9450K .......... .......... .......... .......... .......... 56%  133M 0s\n","  9500K .......... .......... .......... .......... .......... 57%  198M 0s\n","  9550K .......... .......... .......... .......... .......... 57%  138M 0s\n","  9600K .......... .......... .......... .......... .......... 57% 93.1M 0s\n","  9650K .......... .......... .......... .......... .......... 58%  193M 0s\n","  9700K .......... .......... .......... .......... .......... 58%  270M 0s\n","  9750K .......... .......... .......... .......... .......... 58%  231M 0s\n","  9800K .......... .......... .......... .......... .......... 58%  210M 0s\n","  9850K .......... .......... .......... .......... .......... 59%  126M 0s\n","  9900K .......... .......... .......... .......... .......... 59%  171M 0s\n","  9950K .......... .......... .......... .......... .......... 59%  114M 0s\n"," 10000K .......... .......... .......... .......... .......... 60%  111M 0s\n"," 10050K .......... .......... .......... .......... .......... 60% 34.1M 0s\n"," 10100K .......... .......... .......... .......... .......... 60% 32.3M 0s\n"," 10150K .......... .......... .......... .......... .......... 61% 44.0M 0s\n"," 10200K .......... .......... .......... .......... .......... 61% 31.3M 0s\n"," 10250K .......... .......... .......... .......... .......... 61% 44.8M 0s\n"," 10300K .......... .......... .......... .......... .......... 61% 45.1M 0s\n"," 10350K .......... .......... .......... .......... .......... 62% 45.4M 0s\n"," 10400K .......... .......... .......... .......... .......... 62% 43.0M 0s\n"," 10450K .......... .......... .......... .......... .......... 62% 38.1M 0s\n"," 10500K .......... .......... .......... .......... .......... 63% 46.3M 0s\n"," 10550K .......... .......... .......... .......... .......... 63% 23.1M 0s\n"," 10600K .......... .......... .......... .......... .......... 63% 38.4M 0s\n"," 10650K .......... .......... .......... .......... .......... 64% 45.2M 0s\n"," 10700K .......... .......... .......... .......... .......... 64% 45.0M 0s\n"," 10750K .......... .......... .......... .......... .......... 64% 40.7M 0s\n"," 10800K .......... .......... .......... .......... .......... 64% 43.6M 0s\n"," 10850K .......... .......... .......... .......... .......... 65%  241M 0s\n"," 10900K .......... .......... .......... .......... .......... 65%  277M 0s\n"," 10950K .......... .......... .......... .......... .......... 65%  251M 0s\n"," 11000K .......... .......... .......... .......... .......... 66%  183M 0s\n"," 11050K .......... .......... .......... .......... .......... 66%  221M 0s\n"," 11100K .......... .......... .......... .......... .......... 66%  211M 0s\n"," 11150K .......... .......... .......... .......... .......... 67%  197M 0s\n"," 11200K .......... .......... .......... .......... .......... 67%  185M 0s\n"," 11250K .......... .......... .......... .......... .......... 67%  218M 0s\n"," 11300K .......... .......... .......... .......... .......... 67%  228M 0s\n"," 11350K .......... .......... .......... .......... .......... 68%  251M 0s\n"," 11400K .......... .......... .......... .......... .......... 68%  167M 0s\n"," 11450K .......... .......... .......... .......... .......... 68%  246M 0s\n"," 11500K .......... .......... .......... .......... .......... 69%  205M 0s\n"," 11550K .......... .......... .......... .......... .......... 69%  229M 0s\n"," 11600K .......... .......... .......... .......... .......... 69%  196M 0s\n"," 11650K .......... .......... .......... .......... .......... 70%  250M 0s\n"," 11700K .......... .......... .......... .......... .......... 70% 85.8M 0s\n"," 11750K .......... .......... .......... .......... .......... 70% 45.0M 0s\n"," 11800K .......... .......... .......... .......... .......... 70% 39.6M 0s\n"," 11850K .......... .......... .......... .......... .......... 71% 44.9M 0s\n"," 11900K .......... .......... .......... .......... .......... 71% 44.8M 0s\n"," 11950K .......... .......... .......... .......... .......... 71% 45.4M 0s\n"," 12000K .......... .......... .......... .......... .......... 72% 37.1M 0s\n"," 12050K .......... .......... .......... .......... .......... 72% 45.9M 0s\n"," 12100K .......... .......... .......... .......... .......... 72% 45.5M 0s\n"," 12150K .......... .......... .......... .......... .......... 73% 45.8M 0s\n"," 12200K .......... .......... .......... .......... .......... 73% 39.7M 0s\n"," 12250K .......... .......... .......... .......... .......... 73% 66.8M 0s\n"," 12300K .......... .......... .......... .......... .......... 73%  259M 0s\n"," 12350K .......... .......... .......... .......... .......... 74%  254M 0s\n"," 12400K .......... .......... .......... .......... .......... 74%  203M 0s\n"," 12450K .......... .......... .......... .......... .......... 74%  239M 0s\n"," 12500K .......... .......... .......... .......... .......... 75%  280M 0s\n"," 12550K .......... .......... .......... .......... .......... 75%  255M 0s\n"," 12600K .......... .......... .......... .......... .......... 75% 11.0M 0s\n"," 12650K .......... .......... .......... .......... .......... 76% 39.3M 0s\n"," 12700K .......... .......... .......... .......... .......... 76% 40.1M 0s\n"," 12750K .......... .......... .......... .......... .......... 76% 52.6M 0s\n"," 12800K .......... .......... .......... .......... .......... 76% 68.7M 0s\n"," 12850K .......... .......... .......... .......... .......... 77%  244M 0s\n"," 12900K .......... .......... .......... .......... .......... 77% 48.5M 0s\n"," 12950K .......... .......... .......... .......... .......... 77%  187M 0s\n"," 13000K .......... .......... .......... .......... .......... 78% 75.3M 0s\n"," 13050K .......... .......... .......... .......... .......... 78% 46.8M 0s\n"," 13100K .......... .......... .......... .......... .......... 78%  164M 0s\n"," 13150K .......... .......... .......... .......... .......... 79%  248M 0s\n"," 13200K .......... .......... .......... .......... .......... 79% 39.4M 0s\n"," 13250K .......... .......... .......... .......... .......... 79%  103M 0s\n"," 13300K .......... .......... .......... .......... .......... 79%  247M 0s\n"," 13350K .......... .......... .......... .......... .......... 80% 68.7M 0s\n"," 13400K .......... .......... .......... .......... .......... 80% 52.3M 0s\n"," 13450K .......... .......... .......... .......... .......... 80%  242M 0s\n"," 13500K .......... .......... .......... .......... .......... 81%  254M 0s\n"," 13550K .......... .......... .......... .......... .......... 81% 53.5M 0s\n"," 13600K .......... .......... .......... .......... .......... 81% 43.1M 0s\n"," 13650K .......... .......... .......... .......... .......... 82% 74.1M 0s\n"," 13700K .......... .......... .......... .......... .......... 82% 84.8M 0s\n"," 13750K .......... .......... .......... .......... .......... 82% 85.3M 0s\n"," 13800K .......... .......... .......... .......... .......... 82%  220M 0s\n"," 13850K .......... .......... .......... .......... .......... 83% 55.3M 0s\n"," 13900K .......... .......... .......... .......... .......... 83% 98.5M 0s\n"," 13950K .......... .......... .......... .......... .......... 83%  233M 0s\n"," 14000K .......... .......... .......... .......... .......... 84%  222M 0s\n"," 14050K .......... .......... .......... .......... .......... 84%  267M 0s\n"," 14100K .......... .......... .......... .......... .......... 84%  243M 0s\n"," 14150K .......... .......... .......... .......... .......... 85%  233M 0s\n"," 14200K .......... .......... .......... .......... .......... 85%  214M 0s\n"," 14250K .......... .......... .......... .......... .......... 85%  253M 0s\n"," 14300K .......... .......... .......... .......... .......... 85%  266M 0s\n"," 14350K .......... .......... .......... .......... .......... 86%  228M 0s\n"," 14400K .......... .......... .......... .......... .......... 86%  215M 0s\n"," 14450K .......... .......... .......... .......... .......... 86%  268M 0s\n"," 14500K .......... .......... .......... .......... .......... 87%  267M 0s\n"," 14550K .......... .......... .......... .......... .......... 87%  261M 0s\n"," 14600K .......... .......... .......... .......... .......... 87%  201M 0s\n"," 14650K .......... .......... .......... .......... .......... 87%  247M 0s\n"," 14700K .......... .......... .......... .......... .......... 88%  263M 0s\n"," 14750K .......... .......... .......... .......... .......... 88%  259M 0s\n"," 14800K .......... .......... .......... .......... .......... 88%  219M 0s\n"," 14850K .......... .......... .......... .......... .......... 89%  216M 0s\n"," 14900K .......... .......... .......... .......... .......... 89%  198M 0s\n"," 14950K .......... .......... .......... .......... .......... 89%  249M 0s\n"," 15000K .......... .......... .......... .......... .......... 90%  229M 0s\n"," 15050K .......... .......... .......... .......... .......... 90%  273M 0s\n"," 15100K .......... .......... .......... .......... .......... 90% 54.0M 0s\n"," 15150K .......... .......... .......... .......... .......... 90% 49.7M 0s\n"," 15200K .......... .......... .......... .......... .......... 91% 43.0M 0s\n"," 15250K .......... .......... .......... .......... .......... 91% 44.6M 0s\n"," 15300K .......... .......... .......... .......... .......... 91% 41.3M 0s\n"," 15350K .......... .......... .......... .......... .......... 92% 46.0M 0s\n"," 15400K .......... .......... .......... .......... .......... 92% 40.5M 0s\n"," 15450K .......... .......... .......... .......... .......... 92% 48.0M 0s\n"," 15500K .......... .......... .......... .......... .......... 93% 7.71M 0s\n"," 15550K .......... .......... .......... .......... .......... 93% 56.8M 0s\n"," 15600K .......... .......... .......... .......... .......... 93% 50.8M 0s\n"," 15650K .......... .......... .......... .......... .......... 93% 58.6M 0s\n"," 15700K .......... .......... .......... .......... .......... 94%  102M 0s\n"," 15750K .......... .......... .......... .......... .......... 94%  284M 0s\n"," 15800K .......... .......... .......... .......... .......... 94%  265M 0s\n"," 15850K .......... .......... .......... .......... .......... 95%  300M 0s\n"," 15900K .......... .......... .......... .......... .......... 95%  305M 0s\n"," 15950K .......... .......... .......... .......... .......... 95%  308M 0s\n"," 16000K .......... .......... .......... .......... .......... 96%  226M 0s\n"," 16050K .......... .......... .......... .......... .......... 96%  269M 0s\n"," 16100K .......... .......... .......... .......... .......... 96%  253M 0s\n"," 16150K .......... .......... .......... .......... .......... 96%  275M 0s\n"," 16200K .......... .......... .......... .......... .......... 97%  225M 0s\n"," 16250K .......... .......... .......... .......... .......... 97% 44.5M 0s\n"," 16300K .......... .......... .......... .......... .......... 97% 43.4M 0s\n"," 16350K .......... .......... .......... .......... .......... 98% 45.3M 0s\n"," 16400K .......... .......... .......... .......... .......... 98% 38.9M 0s\n"," 16450K .......... .......... .......... .......... .......... 98% 47.8M 0s\n"," 16500K .......... .......... .......... .......... .......... 99% 44.5M 0s\n"," 16550K .......... .......... .......... .......... .......... 99% 98.2M 0s\n"," 16600K .......... .......... .......... .......... .......... 99%  196M 0s\n"," 16650K .......... .......... .......... .......... .......... 99%  250M 0s\n"," 16700K .....                                                 100%  131M=0.4s\n","\n","2022-06-29 16:45:41 (42.4 MB/s) - ‘SPAdes-3.12.0-Linux.tar.gz’ saved [17106384/17106384]\n","\n","bwt_gen.c: In function ‘BWTIncBuildRelativeRank’:\n","bwt_gen.c:879:10: warning: variable ‘oldInverseSa0RelativeRank’ set but not used [-Wunused-but-set-variable]\n","  bgint_t oldInverseSa0RelativeRank = 0;\n","          ^~~~~~~~~~~~~~~~~~~~~~~~~\n","bwt_gen.c: In function ‘BWTIncMergeBwt’:\n","bwt_gen.c:953:15: warning: variable ‘bitsInWordMinusBitPerChar’ set but not used [-Wunused-but-set-variable]\n","  unsigned int bitsInWordMinusBitPerChar;\n","               ^~~~~~~~~~~~~~~~~~~~~~~~~\n","rope.c: In function ‘rope_restore_node’:\n","rope.c:282:2: warning: ignoring return value of ‘fread’, declared with attribute warn_unused_result [-Wunused-result]\n","  fread(&is_bottom, 1, 1, fp);\n","  ^~~~~~~~~~~~~~~~~~~~~~~~~~~\n","rope.c:283:2: warning: ignoring return value of ‘fread’, declared with attribute warn_unused_result [-Wunused-result]\n","  fread(&n, 2, 1, fp);\n","  ^~~~~~~~~~~~~~~~~~~\n","rope.c:291:4: warning: ignoring return value of ‘fread’, declared with attribute warn_unused_result [-Wunused-result]\n","    fread(p[i].c, 8, 6, fp);\n","    ^~~~~~~~~~~~~~~~~~~~~~~\n","rope.c:292:4: warning: ignoring return value of ‘fread’, declared with attribute warn_unused_result [-Wunused-result]\n","    fread(q, 2, 1, fp);\n","    ^~~~~~~~~~~~~~~~~~\n","rope.c:293:4: warning: ignoring return value of ‘fread’, declared with attribute warn_unused_result [-Wunused-result]\n","    fread(q + 1, 1, *q, fp);\n","    ^~~~~~~~~~~~~~~~~~~~~~~\n","rope.c: In function ‘rope_restore’:\n","rope.c:312:2: warning: ignoring return value of ‘fread’, declared with attribute warn_unused_result [-Wunused-result]\n","  fread(&r->max_nodes, 4, 1, fp);\n","  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n","rope.c:313:2: warning: ignoring return value of ‘fread’, declared with attribute warn_unused_result [-Wunused-result]\n","  fread(&r->block_len, 4, 1, fp);\n","  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n","ar: `u' modifier ignored since `D' is the default (see `U')\n","bwashm.c: In function ‘bwa_shm_stage’:\n","bwashm.c:29:2: warning: ignoring return value of ‘ftruncate’, declared with attribute warn_unused_result [-Wunused-result]\n","  ftruncate(shmid, BWA_CTL_SIZE);\n","  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n","bwashm.c:66:2: warning: ignoring return value of ‘ftruncate’, declared with attribute warn_unused_result [-Wunused-result]\n","  ftruncate(shmid, idx->l_mem);\n","  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~\n","bash: line 23: configure: command not found\n","Makefile:98: config.mk: No such file or directory\n","kprobaln.c: In function ‘kpa_glocal’:\n","kprobaln.c:80:21: warning: variable ‘is_diff’ set but not used [-Wunused-but-set-variable]\n","  int bw, bw2, i, k, is_diff = 0, is_backward = 1, Pr;\n","                     ^~~~~~~\n","cut_target.c: In function ‘process_cns.isra.0’:\n","cut_target.c:125:5: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]\n","     for (i = 0, s = -1; i <= l; ++i) {\n","     ^~~\n","ar: `u' modifier ignored since `D' is the default (see `U')\n","misc/ace2sam.c: In function ‘main’:\n","misc/ace2sam.c:164:13: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]\n","             if (write_cns) puts(t[4].s); t[4].l = 0;\n","             ^~\n","misc/ace2sam.c:164:42: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’\n","             if (write_cns) puts(t[4].s); t[4].l = 0;\n","                                          ^\n"]}]},{"cell_type":"code","metadata":{"id":"HNEHJhfdASmy","cellView":"form","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1656521205605,"user_tz":240,"elapsed":6598,"user":{"displayName":"Arav Bhardwaj","userId":"05330003763859094861"}},"outputId":"61ea799a-ee86-448e-e424-360bbb11c555"},"source":["#@title # Set up notebook\n","# import gdown\n","# paths = {'SRR11140744_ILLUMINA.fastq': 'https://drive.google.com/uc?id=1WQ-NS8yloaAORpUwTBYA_Sx2kQ7jhhSK',\n","#          'SRR10948474_OXFORD_NANOPORE.fastq': 'https://drive.google.com/uc?id=1y7B2d8ie8o_hB0kvJXEgevaDv4TOgVmX',\n","#          'NC_045512.2.fasta': 'https://drive.google.com/uc?id=1T9en64lqJfJLMX4dLykYeT1GKhEYoBD8',\n","#          'NC_045512.2.fasta.amb':'https://drive.google.com/uc?id=1fKRqfhMRK3H6Nx4gCpCf85ihf7EXfjF1',\n","#          'NC_045512.2.fasta.ann':'https://drive.google.com/uc?id=1nGhnNOmt46T_ucW0if68gtAqQSXrPsf5',\n","#          'NC_045512.2.fasta.bwt':'https://drive.google.com/uc?id=1ZRR1mwNDl11QHXAjkLLLXmowtW5teUeT',\n","#          'NC_045512.2.fasta.pac':'https://drive.google.com/uc?id=1cBvy8uRj7zrbcsGtQM4TWsrlbDgkYmCk',\n","#          'NC_045512.2.fasta.sa':'https://drive.google.com/uc?id=10QaiPPcwyY66DAGDgzjOBbc22GTdzczm',\n","# }\n","\n","# for k in paths.keys():\n","#   gdown.download(paths[k], 'data/' + k, True)\n","\n","!wget -P 'data' 'https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/SRR11140744_ILLUMINA.fastq'\n","!wget -P 'data' 'https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/SRR10948474_OXFORD_NANOPORE.fastq'\n","!wget -P 'data' 'https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/NC_045512.2.fasta'\n","!wget -P 'data' 'https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/NC_045512.2.fasta.amb'\n","!wget -P 'data' 'https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/NC_045512.2.fasta.ann'\n","!wget -P 'data' 'https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/NC_045512.2.fasta.bwt'\n","!wget -P 'data' 'https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/NC_045512.2.fasta.pac'\n","!wget -P 'data' 'https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/NC_045512.2.fasta.sa'\n"],"execution_count":2,"outputs":[{"output_type":"stream","name":"stdout","text":["--2022-06-29 16:46:36--  https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/SRR11140744_ILLUMINA.fastq\n","Resolving storage.googleapis.com (storage.googleapis.com)... 64.233.188.128, 64.233.189.128, 108.177.125.128, ...\n","Connecting to storage.googleapis.com (storage.googleapis.com)|64.233.188.128|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 347466108 (331M) [application/octet-stream]\n","Saving to: ‘data/SRR11140744_ILLUMINA.fastq’\n","\n","SRR11140744_ILLUMIN 100%[===================>] 331.37M  68.3MB/s    in 4.9s    \n","\n","2022-06-29 16:46:41 (68.3 MB/s) - ‘data/SRR11140744_ILLUMINA.fastq’ saved [347466108/347466108]\n","\n","--2022-06-29 16:46:41--  https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/SRR10948474_OXFORD_NANOPORE.fastq\n","Resolving storage.googleapis.com (storage.googleapis.com)... 74.125.204.128, 64.233.188.128, 64.233.189.128, ...\n","Connecting to storage.googleapis.com (storage.googleapis.com)|74.125.204.128|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 56443872 (54M) [application/octet-stream]\n","Saving to: ‘data/SRR10948474_OXFORD_NANOPORE.fastq’\n","\n","SRR10948474_OXFORD_ 100%[===================>]  53.83M   247MB/s    in 0.2s    \n","\n","2022-06-29 16:46:42 (247 MB/s) - ‘data/SRR10948474_OXFORD_NANOPORE.fastq’ saved [56443872/56443872]\n","\n","--2022-06-29 16:46:42--  https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/NC_045512.2.fasta\n","Resolving storage.googleapis.com (storage.googleapis.com)... 142.250.157.128, 142.251.8.128, 74.125.23.128, ...\n","Connecting to storage.googleapis.com (storage.googleapis.com)|142.250.157.128|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 30429 (30K) [application/octet-stream]\n","Saving to: ‘data/NC_045512.2.fasta’\n","\n","NC_045512.2.fasta   100%[===================>]  29.72K  --.-KB/s    in 0s      \n","\n","2022-06-29 16:46:42 (120 MB/s) - ‘data/NC_045512.2.fasta’ saved [30429/30429]\n","\n","--2022-06-29 16:46:42--  https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/NC_045512.2.fasta.amb\n","Resolving storage.googleapis.com (storage.googleapis.com)... 142.250.157.128, 142.251.8.128, 74.125.23.128, ...\n","Connecting to storage.googleapis.com (storage.googleapis.com)|142.250.157.128|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 10 [application/octet-stream]\n","Saving to: ‘data/NC_045512.2.fasta.amb’\n","\n","NC_045512.2.fasta.a 100%[===================>]      10  --.-KB/s    in 0s      \n","\n","2022-06-29 16:46:42 (1.54 MB/s) - ‘data/NC_045512.2.fasta.amb’ saved [10/10]\n","\n","--2022-06-29 16:46:42--  https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/NC_045512.2.fasta.ann\n","Resolving storage.googleapis.com (storage.googleapis.com)... 74.125.204.128, 64.233.187.128, 64.233.189.128, ...\n","Connecting to storage.googleapis.com (storage.googleapis.com)|74.125.204.128|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 119 [application/octet-stream]\n","Saving to: ‘data/NC_045512.2.fasta.ann’\n","\n","NC_045512.2.fasta.a 100%[===================>]     119  --.-KB/s    in 0s      \n","\n","2022-06-29 16:46:42 (31.0 MB/s) - ‘data/NC_045512.2.fasta.ann’ saved [119/119]\n","\n","--2022-06-29 16:46:42--  https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/NC_045512.2.fasta.bwt\n","Resolving storage.googleapis.com (storage.googleapis.com)... 74.125.23.128, 74.125.203.128, 74.125.204.128, ...\n","Connecting to storage.googleapis.com (storage.googleapis.com)|74.125.23.128|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 30000 (29K) [application/octet-stream]\n","Saving to: ‘data/NC_045512.2.fasta.bwt’\n","\n","NC_045512.2.fasta.b 100%[===================>]  29.30K  --.-KB/s    in 0s      \n","\n","2022-06-29 16:46:42 (107 MB/s) - ‘data/NC_045512.2.fasta.bwt’ saved [30000/30000]\n","\n","--2022-06-29 16:46:42--  https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/NC_045512.2.fasta.pac\n","Resolving storage.googleapis.com (storage.googleapis.com)... 74.125.23.128, 74.125.203.128, 74.125.204.128, ...\n","Connecting to storage.googleapis.com (storage.googleapis.com)|74.125.23.128|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 7477 (7.3K) [application/octet-stream]\n","Saving to: ‘data/NC_045512.2.fasta.pac’\n","\n","NC_045512.2.fasta.p 100%[===================>]   7.30K  --.-KB/s    in 0s      \n","\n","2022-06-29 16:46:42 (83.2 MB/s) - ‘data/NC_045512.2.fasta.pac’ saved [7477/7477]\n","\n","--2022-06-29 16:46:42--  https://storage.googleapis.com/inspirit-ai-data-bucket-1/Data/AI%20Scholars/Sessions%206%20-%2010%20(Projects)/Project%20-%20DNA%20Detectives/NC_045512.2.fasta.sa\n","Resolving storage.googleapis.com (storage.googleapis.com)... 74.125.23.128, 74.125.203.128, 74.125.204.128, ...\n","Connecting to storage.googleapis.com (storage.googleapis.com)|74.125.23.128|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 15000 (15K) [application/octet-stream]\n","Saving to: ‘data/NC_045512.2.fasta.sa’\n","\n","NC_045512.2.fasta.s 100%[===================>]  14.65K  --.-KB/s    in 0s      \n","\n","2022-06-29 16:46:42 (107 MB/s) - ‘data/NC_045512.2.fasta.sa’ saved [15000/15000]\n","\n"]}]},{"cell_type":"markdown","metadata":{"id":"Mv4msqNQhqfw"},"source":["# Sequencing"]},{"cell_type":"markdown","metadata":{"id":"aDIjpc7rL3Ia"},"source":["***Sequencing*** is the process of reading the DNA or RNA of an organism.  There are several types of sequencing machines that take advantage of the unique chemical properties of A, T, C, and G to read through a piece of piece of DNA and learn it's bases.\n","\n","There are two main types of sequencing: sequencing by synthesis and nanopore-based sequencing.\n","\n"]},{"cell_type":"markdown","metadata":{"id":"We3htOirNyYo"},"source":["## Sequencing by Synthesis\n","\n","In sequencing-by-synthesis, a special enzyme adds glowing complementary nucleotides to a piece of DNA that are then read by a camera.\n","\n","[![](http://img.youtube.com/vi/3M0PyxFPwkQ/0.jpg)](https://www.youtube.com/watch?v=3M0PyxFPwkQ \"Sequencing by Synthesis\")"]},{"cell_type":"markdown","metadata":{"id":"J8C2kbdKN2HC"},"source":["## Nanopore-based Sequencing\n","\n","\n","In nanopore-based sequencing, DNA is pulled through a special pore that generates a current when different bases go through the pore. The current is read by a voltmeter and converted into A,T,C, and G.\n","\n","Click on the image below to watch an animation of nanopore-based sequencing.\n","\n","[![](http://img.youtube.com/vi/RcP85JHLmnI/0.jpg)](https://www.youtube.com/watch?v=RcP85JHLmnI \"Nanopore\")"]},{"cell_type":"markdown","metadata":{"id":"4DWsl34bD_0G"},"source":["## Building Genomes"]},{"cell_type":"markdown","metadata":{"id":"uZ2MjiLReLv4"},"source":["\n","In the sequencing pipeline, DNA is first broken up into many small segments. Then the sequencer reads one segment at at time. (For reference, Sanger sequencing reads are often < 200 basepairs long, and the very first human genome sequenced (3 billion basepairs long) was done using Sanger sequencing!)\n","\n","**How do we go from these small sequences to knowing the full genome of an organism or sample?**\n","\n","There are two main approaches: *De novo* assembly and reference-based alignment. *De novo* assembly (you have no idea what the genome looks like) is like putting together a jigsaw puzzle with no photo of what it should look like in the end). Reference-based assembly (you know what the genome should look like and you want to find small variations in your sample) is like putting together a jigsaw puzzle while looking at the box of what the finished product should look like.\n","\n","\n","\n","![De Novo vs Reference-Based Assembly](https://www.researchgate.net/profile/Christopher_Noune/publication/321179957/figure/fig4/AS:614377665343516@1523490458663/6-Comparison-of-reference-assembly-and-de-novo-assembly-A-Reference-assembly-maps.png)\n"]},{"cell_type":"markdown","metadata":{"id":"wlRvoacXEKOd"},"source":["## Reference-based Alignment\n"]},{"cell_type":"markdown","metadata":{"id":"pkJEZpA9ENON"},"source":["In reference-based alignment, we know what the genome of an organism theoretically looks like. Our goal is to map reads back to the part of the genome they belong to. We can then use the aligned reads to count how many reads aligned to which regions of the genome, to measure things like gene expression. Or we can identify regions of a genome in which our sample may have mutations."]},{"cell_type":"markdown","metadata":{"id":"hDilHnhhEFLn"},"source":["## De novo assembly of an organism\n"]},{"cell_type":"markdown","metadata":{"id":"3PnJaLNKEH9O"},"source":["In de novo assembly of an genome, we have no idea what the genome is! We must build it entirely from scratch. In the SARS-CoV-2 (the virus that causes COVID-19), the scientists in Wuhan who first discovered the virus performed [de novo assembly of the virus](https://pubmed.ncbi.nlm.nih.gov/32004165/). They took a throat swab from a patient. They sequenced *everything* in the swab. They threw out any reads that looked like human DNA (see reference-based assembly for how they did this). Then they pieced together the rest of the reads to produce the SARS-CoV-2 genome.\n","\n","De novo assembly works best with the longest length of reads possible. Oxford nanopore is the hottest new technology for this purpose. In de novo assembly, we look for some amount of overlap between different reads and then overlay them. We keep doing this until we have 1 or more long lengths of unbroken sequences (also known as contigs). \n"]},{"cell_type":"markdown","metadata":{"id":"-Qu5MQerPCct"},"source":["## **Exercise: Manually perform de novo asssembly and reference-based alignment from the reads in the activity slide deck.**\n","\n","Make a copy of ```5_Bonus_Genome_Assembly_Activity```, open up the slide deck and move around the reads in order to perform de novo assembly and reference based alignment.\n","\n","Record the mutations you found in your reference based alignment, and the sequence in your de novo assembly."]},{"cell_type":"code","metadata":{"id":"HTsZvX2OP1Me","cellView":"form","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1656522233377,"user_tz":240,"elapsed":292,"user":{"displayName":"Arav Bhardwaj","userId":"05330003763859094861"}},"outputId":"2f480986-61ff-4570-c9ed-d071f6ac72b6"},"source":["mutation_location  =  '14' #@param {type:\"string\"}\n","mutation_base  =  'G switched to T' #@param {type:\"string\"}\n","de_novo_sequence  =  'CGGGTAGCCACTGACGTAGCCGGTTGTCCACTT' #@param {type:\"string\"}\n","\n","print('Mutation Location: 14th position')\n","print(\"Mutation Base: G switched to T\")\n","print(\"De novo sequence: CGGGTAGCCACTGACGTAGCCGGTTGTCCACTT\")"],"execution_count":3,"outputs":[{"output_type":"stream","name":"stdout","text":["Mutation Location: 14th position\n","Mutation Base: G switched to T\n","De novo sequence: CGGGTAGCCACTGACGTAGCCGGTTGTCCACTT\n"]}]},{"cell_type":"markdown","metadata":{"id":"-wpDvTGHEP1o"},"source":["---"]},{"cell_type":"markdown","metadata":{"id":"j5ENumTEqXmU"},"source":["# Reference-Based Alignment -- Do it yourself!"]},{"cell_type":"markdown","metadata":{"id":"_UtTPoZrBTx_"},"source":["Situation: We know what the genome of an organism *should* look like. We sequence a sample and we want to know how it's genome compared to the theoretical genome. What are some examples of situtations where this might be the case?\n","\n"]},{"cell_type":"markdown","metadata":{"id":"WqXEuzAuPtZL"},"source":["The SARS-CoV-2 virus is actively mutating, so are curious which mutations a specific evolutionary lineage of SARS-CoV-2 has accumulated. We sequence the SARS-CoV-2 strain of interest (this one is from Australia) and compare it to our reference genome to find mutations."]},{"cell_type":"markdown","metadata":{"id":"X6a9ziyILe5r"},"source":["### What we need\n","- The reads from the sequencer (fastq files).\n","- One or more [reference genomes](https://www.ncbi.nlm.nih.gov/nuccore/NC_045512.2) (fasta files)."]},{"cell_type":"markdown","metadata":{"id":"EUW4gQ_JQ39A"},"source":["## Step 1: Examine your reference genome(s).\n","\n","We are using [this reference genome](https://www.ncbi.nlm.nih.gov/nuccore/NC_045512.2) (one of the first samples of SARS-CoV-2). What does a reference genome look like?"]},{"cell_type":"code","metadata":{"id":"2tiIqiWsKcU-","cellView":"form","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1656522241356,"user_tz":240,"elapsed":298,"user":{"displayName":"Arav Bhardwaj","userId":"05330003763859094861"}},"outputId":"a505fd67-fc7f-497c-9d9b-c2d8c105237f"},"source":["#@title #### **Exercise: Examine your reference genome.**\n","%%bash\n","\n","# Show us what the reference genome file looks like (first 30 lines).\n","head -n 30 data/NC_045512.2.fasta"],"execution_count":4,"outputs":[{"output_type":"stream","name":"stdout","text":[">NC_045512.2 Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome\n","ATTAAAGGTTTATACCTTCCCAGGTAACAAACCAACCAACTTTCGATCTCTTGTAGATCTGTTCTCTAAA\n","CGAACTTTAAAATCTGTGTGGCTGTCACTCGGCTGCATGCTTAGTGCACTCACGCAGTATAATTAATAAC\n","TAATTACTGTCGTTGACAGGACACGAGTAACTCGTCTATCTTCTGCAGGCTGCTTACGGTTTCGTCCGTG\n","TTGCAGCCGATCATCAGCACATCTAGGTTTCGTCCGGGTGTGACCGAAAGGTAAGATGGAGAGCCTTGTC\n","CCTGGTTTCAACGAGAAAACACACGTCCAACTCAGTTTGCCTGTTTTACAGGTTCGCGACGTGCTCGTAC\n","GTGGCTTTGGAGACTCCGTGGAGGAGGTCTTATCAGAGGCACGTCAACATCTTAAAGATGGCACTTGTGG\n","CTTAGTAGAAGTTGAAAAAGGCGTTTTGCCTCAACTTGAACAGCCCTATGTGTTCATCAAACGTTCGGAT\n","GCTCGAACTGCACCTCATGGTCATGTTATGGTTGAGCTGGTAGCAGAACTCGAAGGCATTCAGTACGGTC\n","GTAGTGGTGAGACACTTGGTGTCCTTGTCCCTCATGTGGGCGAAATACCAGTGGCTTACCGCAAGGTTCT\n","TCTTCGTAAGAACGGTAATAAAGGAGCTGGTGGCCATAGTTACGGCGCCGATCTAAAGTCATTTGACTTA\n","GGCGACGAGCTTGGCACTGATCCTTATGAAGATTTTCAAGAAAACTGGAACACTAAACATAGCAGTGGTG\n","TTACCCGTGAACTCATGCGTGAGCTTAACGGAGGGGCATACACTCGCTATGTCGATAACAACTTCTGTGG\n","CCCTGATGGCTACCCTCTTGAGTGCATTAAAGACCTTCTAGCACGTGCTGGTAAAGCTTCATGCACTTTG\n","TCCGAACAACTGGACTTTATTGACACTAAGAGGGGTGTATACTGCTGCCGTGAACATGAGCATGAAATTG\n","CTTGGTACACGGAACGTTCTGAAAAGAGCTATGAATTGCAGACACCTTTTGAAATTAAATTGGCAAAGAA\n","ATTTGACACCTTCAATGGGGAATGTCCAAATTTTGTATTTCCCTTAAATTCCATAATCAAGACTATTCAA\n","CCAAGGGTTGAAAAGAAAAAGCTTGATGGCTTTATGGGTAGAATTCGATCTGTCTATCCAGTTGCGTCAC\n","CAAATGAATGCAACCAAATGTGCCTTTCAACTCTCATGAAGTGTGATCATTGTGGTGAAACTTCATGGCA\n","GACGGGCGATTTTGTTAAAGCCACTTGCGAATTTTGTGGCACTGAGAATTTGACTAAAGAAGGTGCCACT\n","ACTTGTGGTTACTTACCCCAAAATGCTGTTGTTAAAATTTATTGTCCAGCATGTCACAATTCAGAAGTAG\n","GACCTGAGCATAGTCTTGCCGAATACCATAATGAATCTGGCTTGAAAACCATTCTTCGTAAGGGTGGTCG\n","CACTATTGCCTTTGGAGGCTGTGTGTTCTCTTATGTTGGTTGCCATAACAAGTGTGCCTATTGGGTTCCA\n","CGTGCTAGCGCTAACATAGGTTGTAACCATACAGGTGTTGTTGGAGAAGGTTCCGAAGGTCTTAATGACA\n","ACCTTCTTGAAATACTCCAAAAAGAGAAAGTCAACATCAATATTGTTGGTGACTTTAAACTTAATGAAGA\n","GATCGCCATTATTTTGGCATCTTTTTCTGCTTCCACAAGTGCTTTTGTGGAAACTGTGAAAGGTTTGGAT\n","TATAAAGCATTCAAACAAATTGTTGAATCCTGTGGTAATTTTAAAGTTACAAAAGGAAAAGCTAAAAAAG\n","GTGCCTGGAATATTGGTGAACAGAAATCAATACTGAGTCCTCTTTATGCATTTGCATCAGAGGCTGCTCG\n","TGTTGTACGATCAATTTTCTCCCGCACTCTTGAAACTGCTCAAAATTCTGTGCGTGTTTTACAGAAGGCC\n","GCTATAACAATACTAGATGGAATTTCACAGTATTCACTGAGACTCATTGATGCTATGATGTTCACATCTG\n"]}]},{"cell_type":"markdown","metadata":{"id":"ZuyIMGC1Q9KF"},"source":["## Step 2: Examine your reads.\n","\n","We'll be using the reads from a SARS-CoV-2 [sample](https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR11140744) taken from University of Wisconsin - Madison."]},{"cell_type":"code","metadata":{"id":"8tNjwjYjRKGw","cellView":"form","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1656522247310,"user_tz":240,"elapsed":286,"user":{"displayName":"Arav Bhardwaj","userId":"05330003763859094861"}},"outputId":"3dbee055-a677-4284-a535-7225c512458c"},"source":["#@title #### **Exercise: Examine your reads.**\n","%%bash\n","\n","SRR_ID=SRR11140744_ILLUMINA\n","head -n 30 data/${SRR_ID}.fastq"],"execution_count":5,"outputs":[{"output_type":"stream","name":"stdout","text":["@SRR11140744.1/1\n","ACATAGGGCTGTTCAAGTTGAGGCAAAACGCCTTTTTCAACTTCTACTAAGCCACAAGTGCCATCTTTAAGATGTTGACGTGCCTCTGATAAGACCTCCTCCACGGAGTCTCCAAAGCCACGTACGAGCACGTCGCGAACCTGTAAAACAGGCAAACTGAGTTGGACGTGTGTTTTCTCGTTGAAACCAGGGACAAGGCTCTCCATCTTACCTTTCGGTCACACCCGGACGAAACCTAGATGTGCTGATGA\n","+\n","BCCCCFFFFFCFGGGGGGGGGGHGGHHHHGGGHGHHHHHHHHHHHHHHHHHHHHHHHGHHHHHHHHHHHHHHHHHHHHHGGGHHHHHGHHGHHHHHHHHHHHHHGGGGGHHHHHHHHHHHHGHHHGGGGGHGHHGGGGGGGHHHHHHHHHHHGGHHHHHFHHHHHHHGGGHHHHHHHHHGGGHHHHHHHHGGGGGGGFGGGGGGGGGGGGGGGGGGGGGGFFFFFFFFFDFFFFFFFFFFFFFFFFFFFFB\n","@SRR11140744.1/2\n","ACAGGACACGAGTAACTCGTCTATCTTCTGCTGGCTGCTTACGGTTTCGTCCGTGTTGCAGCCGATCATCAGCACATCTAGGTTTCGTCCGGGTGTGACCGAAAGGTAAGATGGAGAGCCTTGTCCCTGGTTTCAACGAGAAAACACACGTCCAACTCAGTTTGCCTGTTTTACAGGTTCGCGACGTGCTCGTACGTGGCTTTGGAGACTCCGTGGAGGAGGTCTTATCAGAGGCACGTCAACATCTTAAA\n","+\n","ABAAAFBFFBDBGGGGGGGGGGHHHHHHHHHHCHGHGGGHHHGGHGGHGHGGGHFHHHHHHHHGGGGGHHHHHHHHHFHHHHGHHHGHGGGGGEFGDGHHGFGGGHHHHHGHHGGHHFHHHHGHHHHHHHHHHHHHHGFFGGHHHHHHGGHHGGHHHHHEGHHHHHHHGHHGHHFHHHHHGGGGGGGGGGGGAGGG9BEFFFFFFFFFFFFFFEEFFFFFFFA.FFFFFFFEFEFFFFFFF.BFFFFFFFB\n","@SRR11140744.2/1\n","CGTACGAGCACGTCGCGAACCTGTAAAACAGGCAAACTGAGTTGGACGTGTGTTTTCTCGTTGAAACCAGGGACAAGGCTCTCCATCTTACCTTTCGGTCACACCCGGACGAAACCTAGATGTGCTGATGATCGGCTGCAACACGGACGAAACCGTAAGCAGCCTGCAGAAGATAGACGAGTTACTCGTGTCCTGTCAACGACAGTAATTAGTTATTAATTATACTGCGTGAGTGCAC\n","+\n","AA>ABCCCCCFCGGGGGGGGGGHHHHHHHHHHGHHHHHHHHHHHHHHGFGGHHHHHHHHGGGHHHHHHHHGGGGGHHHHHHHHHHHHHHHHHHHHHGGGGGHHHGGGGGGGGGGGHHHHHHHHHHHHHHHHHHFGGGGGHHHHGGGGGGGGGGHGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\n","@SRR11140744.2/2\n","GTGCACTCACGCAGTATAATTAATAACTAATTACTGTCGTTGACAGGACACGAGTAACTCGTCTATCTTCTGCAGGCTGCTTACGGTTTCGTCCGTGTTGCAGCCGATCATCAGCACATCTAGGTTTCGTCCGGGTGTGACCGAAAGGTAAGATGGAGAGCCTTGTCCCTGGTTTCAACGAGAAAACACACGTCCAACTCAGTTTGCCTGTTTTACAGGTTCGCGACGTGCTCGTACG\n","+\n","AABCBFFFFFCFGGGGGGDGGGHHHHHHHHHHHHHHHHGGGHHHHHGHHHHGGGGHGHHHGHGHHHHHHHHHHHFGGHHGHHHHGGHGGHGHGGGFFHGHHHGHHGGGGGHHHHHHHHHHHHHHGHHHGHGGGGGGGGGDHHGGFGGHDFHHHHHHHGFFHHHHFHHHHHGHHGHHHHHGGGGGHHFHEHGGGGGGGFGGGGFFFGGGGGGGGGGGGFFFFFGFFFFFFFFFFFCDFF\n","@SRR11140744.3/1\n","CACATAGGGCTGTTCAAGTTGAGGCAAAACGCCTTTTTCAACTTCTACTAAGCCACAAGTGCCATCTTTAAGATGTTGACGTGCCTCTGATAAGACCTCCTCCACGGAGTCTCCAAAGCCACGTACGAGCACGTCGCGAACCTGTAAAACAGGCAAACTGAGTTGGACGTGTGTTTTCTCGTTGAAACCAGGGACAAGGCTCTCCATCTTACCTTTCGGTCACACCCGGACGAAACCTAGATGTGCTGATG\n","+\n","CCCCCFFFFFFCGGGGGGGGGGHHGGHHHHGGGHGHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHGGGHHHHHHHHHHHHHHHHHHHHHHGGGGGHHHHHHHGHHHHGHHHGGGGGHGHHGGGGGGGHHHHHHHHHHHGHHHHHHHHHHHHHHGGGHHHHHHHHHGGGHHHHHHHHGFGGGGGGGGGGGGGGGGGGGGGGGGGGGGFFFFFFFFFFFFFFFFFFFFFFFFFFFFFB\n","@SRR11140744.3/2\n","TGACAGGACACGAGTAACTCGTCTATCTTCTGCAGGCTGCTTACGGTTTCGTCCGTGTTGCAGCCGATCATCAGCACATCTAGGTTTCGTCCGGGTGTGACCGAAAGGTAAGATGGAGAGCCTTGTCCCTGGTTTCAACGAGAAAACACACGTCCAACTCAGTTTGCCTGTTTTACAGGTTCGCGACGTGCTCGTACGTGGCTTTGGAGACTCCGTGGAGGAGGTCTTATCAGAGGCACGTCAACATCTT\n","+\n","BBCCCFCFDFFDGGGGGGGGGGGHHHHHHHHHHHHHGHHGHHHHGGHGGHGHGGGHGHHHHHHHHGGGGGHHHHHGHHHHHGHHHHHHGHGGGGGEFGGHHHGGGGGHHHHHHHHHHHHGHHHHHHHHHHHHHHHHHHGGGGGFHHHHHHGGHHGGHHHHHHHHHHHHHHHHHHHHFHHHHHGGGGGGGGFGGGGGGA:FECFFFFFFFBEFFFFFFFFFFDDDDFFB0FFFFFFFFFFFFFFFFFFFFF\n","@SRR11140744.4/1\n","CTTTAAGATGTTGACGTGCCTCTGATAAGACCTCCTCCACGGAGTCTCCAAAGCCACGTACGAGCACGTCGCGAACCTGTAAAACAGGCAAACTGAGTTGGACGTGTGTCTTCTCGTTGAAACCAGGGACAAGGCTCTCCATCTTACCTTTCGGTCACACCCGGACGAAACCTAGATGTGCTGATGATCGGCTGCAACACGGACGAAACCGTAAGCAGCCTGCAGAAGATAGACGAGTTACTCGTGTCC\n","+\n","CCCCCFFFFFFFGGGGGGGGGGHHHHHHFHHHHHHHHHHHGGGGGHHHHHHHHHHHHGHHHGGGGGHGHHGGGGGGGHHHHHHHHHHHGHHHHHHHHHHHHHHGGGHHHHHHHHHGGGHHHHHHHHGGGGGHHHHHHHHHHHHHHHHHHHHHGGGGGHHHGGGGGGGGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGFFFFFFFFFFFAEFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\n","@SRR11140744.4/2\n","GGACACGAGTAACTCGTCTATCTTCTGCAGGCTGCTTACGGTTTCGTCCGTGTTGCAGCCGATCATCAGCACATCTAGGTTTCGTCCGGGTGTGACCGAAAGGTAAGATGGAGAGCCTTGTCCCTGGTTTCAACGAGAAGACACACGTCCAACTCAGTTTGCCTGTTTTACAGGTTCGCGACGTGCTCGTACGTGGCTTTGGAGACTCCGTGGAGGAGGTCTTATCAGAGGCACGTCAACATCTTAAAG\n"]}]},{"cell_type":"markdown","metadata":{"id":"zzpfWAciDqG9"},"source":[" ### Fields to care about.\n"]},{"cell_type":"markdown","metadata":{"id":"Y5bS4sL1XRl-"},"source":[" `@SRR11494554.6`: This is just the READ_ID, used so we can remember which sequence came from which read.\n","\n","`ACCTTGAAGTGTTATCATTAG....`: This is the string of basepairs that the machine output it sequence that piece of DNA.\n","\n","`/AAAAE6EEEEEEEEEEEEEEEEEEEEEEEEEEEEE`: This is the read quality. This is code that the machine outputs to describe how confident it was in each of the basepairs it called.\n","\n","\n"]},{"cell_type":"markdown","metadata":{"id":"WCz7FWU8RPCp"},"source":["## Step 3: Align your reads to the reference genome.\n","\n","\n"]},{"cell_type":"code","metadata":{"id":"9mVPLDQaMzth","cellView":"form"},"source":["#@title #### **Exercise: Align your reads to the reference genome.**\n","\n","%%bash\n","#################################################################\n","##### SETTING UP FILE DIRECTORIES/PERMISSIONS -- IGNORE ########\n","export PATH='/content/software/bwa':$PATH\n","DATA_DIR=data\n","RESULTS_DIR=results\n","#################################################################\n","#################################################################\n","SRR_ID=SRR11140744_ILLUMINA\n","REFERENCE_GENOME=NC_045512.2.fasta\n","cp ${DATA_DIR}/${REFERENCE_GENOME} ${RESULTS_DIR}/${REFERENCE_GENOME}\n","\n","\n","# Align reads to reference genome.\n","bwa index ${RESULTS_DIR}/${REFERENCE_GENOME} # We must index the reference file for bwa to work.\n","echo \"DONE CREATING INDEX.\"\n","\n","# Perform alignment algorithm (burrows-wheeler algorithm).\n","bwa mem ${RESULTS_DIR}/${REFERENCE_GENOME} -verbose -o ${RESULTS_DIR}/${SRR_ID}.bam ${DATA_DIR}/${SRR_ID}.fastq  "],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"jnx6FNTIRWtN"},"source":["## Step 4: Examine your mapped reads."]},{"cell_type":"code","metadata":{"id":"LEFrGvmC39JE","cellView":"form","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1656522276328,"user_tz":240,"elapsed":336,"user":{"displayName":"Arav Bhardwaj","userId":"05330003763859094861"}},"outputId":"d074e2c7-249f-4571-bdf4-16790cccd18e"},"source":["#@title #### **Exercise: Examine your mapped reads.**\n","%%bash\n","#################################################################\n","##### SETTING UP FILE DIRECTORIES/PERMISSIONS -- IGNORE ########\n","SRR_ID=SRR11140744_ILLUMINA\n","#################################################################\n","#################################################################\n","\n","# Split the aligned file into one file, \n","# with reads that aligned to the reference (_mapped),\n","# and one file with reads that did not align (_unmapped).\n","samtools view  -f 0x04 -b -o results/${SRR_ID}_unmapped.bam results/${SRR_ID}.bam\n","samtools view  -F 0x04 -b -o results/${SRR_ID}_mapped.bam results/${SRR_ID}.bam\n","\n","# Report the number of reads that were aligned and not aligned.\n","echo \"# aligned to reference: \" $(samtools view -c results/${SRR_ID}_mapped.bam) # The -c flag means \"count\" the number of reads.\n","echo \"# unaligned to reference: \" $(samtools view -c results/${SRR_ID}_unmapped.bam) # The -c flag means \"count\" the number of reads.\n","\n","# Sort and index the files so that we can easily view them on IGV.\n","samtools sort -o results/${SRR_ID}_mapped_sorted.bam results/${SRR_ID}_mapped.bam \n","samtools index -b results/${SRR_ID}_mapped_sorted.bam results/${SRR_ID}_mapped_sorted.bam.bai\n","\n","# Show us what a .bam file looks like! (Uncompressed)\n","samtools view results/${SRR_ID}_mapped_sorted.bam | head -n 1000 | tail -n 10"],"execution_count":6,"outputs":[{"output_type":"stream","name":"stdout","text":["# aligned to reference: \n","# unaligned to reference: \n"]},{"output_type":"stream","name":"stderr","text":["[E::hts_open_format] fail to open file 'results/SRR11140744_ILLUMINA.bam'\n","samtools view: failed to open \"results/SRR11140744_ILLUMINA.bam\" for reading: No such file or directory\n","[E::hts_open_format] fail to open file 'results/SRR11140744_ILLUMINA.bam'\n","samtools view: failed to open \"results/SRR11140744_ILLUMINA.bam\" for reading: No such file or directory\n","[E::hts_open_format] fail to open file 'results/SRR11140744_ILLUMINA_mapped.bam'\n","samtools view: failed to open \"results/SRR11140744_ILLUMINA_mapped.bam\" for reading: No such file or directory\n","[E::hts_open_format] fail to open file 'results/SRR11140744_ILLUMINA_unmapped.bam'\n","samtools view: failed to open \"results/SRR11140744_ILLUMINA_unmapped.bam\" for reading: No such file or directory\n","[E::hts_open_format] fail to open file 'results/SRR11140744_ILLUMINA_mapped.bam'\n","[bam_sort_core] fail to open 'results/SRR11140744_ILLUMINA_mapped.bam': No such file or directory\n","[E::hts_open_format] fail to open file 'results/SRR11140744_ILLUMINA_mapped_sorted.bam'\n","samtools index: failed to open \"results/SRR11140744_ILLUMINA_mapped_sorted.bam\": No such file or directory\n","[E::hts_open_format] fail to open file 'results/SRR11140744_ILLUMINA_mapped_sorted.bam'\n","samtools view: failed to open \"results/SRR11140744_ILLUMINA_mapped_sorted.bam\" for reading: No such file or directory\n"]}]},{"cell_type":"markdown","metadata":{"id":"CyVgKGD2DgJB"},"source":["### Fields to care about.\n"]},{"cell_type":"markdown","metadata":{"id":"rJ7HxOlUYAs-"},"source":["\n"," `@SRR11140744.3137`: This is the read id.\n","\n","`NC_045512.2`: This is the reference genome (or chromsome) the read mapped to (sometimes we can use a set of multiple genomes as our reference, if we are not sure what organism a read may have come from - example: sequencing the bacteria in your gut.) In this case, we only used one reference genome so everything mapps to NC_045512 (the NCBI code for the SARS-CoV-2 reference sequence).\n","\n","`170`: This is the position the beginning of the read aligned to on the reference genome.\n","\n","`6S226M`: This is something called a CIGAR string that indicates how many mismatches there were between the read and the reference genome where it alined. means \"6 basepairs were **S**oft-clipped (6 bases from one end of the sequence didn't align) and 226 basepairs **M**atched\". Sometimes a read can align, but not perfectly match up to the reference genome, such as when the read has a mutation (or beacause of sequencing error).\n"]},{"cell_type":"markdown","metadata":{"id":"z9jstYqsDb2K"},"source":["## Step 5: Take a look on IGViewer."]},{"cell_type":"code","metadata":{"id":"JFZz1UBRtL09","cellView":"form","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1656522302892,"user_tz":240,"elapsed":21648,"user":{"displayName":"Arav Bhardwaj","userId":"05330003763859094861"}},"outputId":"5b71227b-3147-4f75-9cb6-60c9da3850fa"},"source":["#@title ####Run this cell to copy your data to google drive\n","from google.colab import drive\n","drive.mount('/content/drive')\n","!cp results/SRR11140744_ILLUMINA_mapped_sorted.* '/content/drive/My Drive'"],"execution_count":7,"outputs":[{"output_type":"stream","name":"stdout","text":["Mounted at /content/drive\n","cp: cannot stat 'results/SRR11140744_ILLUMINA_mapped_sorted.*': No such file or directory\n"]}]},{"cell_type":"markdown","metadata":{"id":"y5L5rHR3RgCB"},"source":["***Congrats!*** You made it throught the sometimes painful processes of genome alignment!  \n","\n","(While you're here, go ahead and run steps 1 & 2 of De Novo Assmbly below -- step 2 can take a while, so we'll let it run while we examine our alignments)\n","\n","\n","![WHEE](https://bestanimations.com/Holidays/Fireworks/fireworks/ba-awesome-colorful-fireworks-animated-gif-image-s.gif)\n","\n","You are now ready to see what your mapped reads look like using [IGViewer](https://igv.org/app/).  This is the most fun part of this lab :)\n","1. Go to IGViewer (integrated genome Viewer) webapp (link above).\n","2. Genomes-->SARS-CoV-2. (You could also input your fasta file from the Google Drive, but used the same reference genome for SARS-CoV-2 as IGV. IGV's is already annotated with the ORFs, etc so it is convenient to just use their preset!)\n","3. Track-->Google Drive. Search for ```SRR11140744_ILLUMINA_mapped_sorted``` and select both `SRR11140744_ILLUMINA_mapped.sorted.bam` and `SRR11140744_ILLUMINA_mapped.sorted.bam.bai` to select your final mapping and it's corresponding index file. It may take a couple minutes for IGV to load it.\n","4. Explore!! Understand what you are looking at. You should zoom in and scroll around!\n","\n","### **Exercise: Answer the following questions about the alignment.**\n","\n"]},{"cell_type":"code","metadata":{"id":"VvrBhx6KuA83"},"source":["_1_ =  '' #@param {type:\"string\"}\n","_2_ =  '' #@param {type:\"string\"}\n"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"Y3x2UXQkuTcv"},"source":["### **Exercise: Find and document the following features:**\n","\n","In this particular SARS-CoV-2 sequence, find the location of:\n","1. A conserved single nucleotide polymorphism (SNP, ie a single basepair mutation)?\n","2. A loci where SARS-CoV-2 is highly variable (sometimes has a mutation and sometimes doesn't)?\n","3. A conserved deletion (hint: it is pretty small! You might have to zoom in and search around.)\n"]},{"cell_type":"code","metadata":{"id":"gTjAMtd4ugF4","cellView":"form"},"source":["_1_ =  '' #@param {type:\"string\"}\n","_2_ =  '' #@param {type:\"string\"}\n","_3_ =  '' #@param {type:\"string\"}\n","print(\"1. Check out position 17,198\")\n","print(\"2. Check out position 11,335\")\n","print(\"3. Check out position 20,297\")"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"doH1yU78qgYT"},"source":["# De Novo Assembly: Do it yourself!"]},{"cell_type":"markdown","metadata":{"id":"hA0IgPIkEfgu"},"source":["Now, you will have the chance to perform de novo assembly and reference-based alignment and then view the SARS-CoV-2 genome, via the same state-of-the-art tools that researchers use to perform genome assembly and alignment."]},{"cell_type":"markdown","metadata":{"id":"CT9zlotY0jsw"},"source":["What we neeed:\n","\n","- Just the raw reads."]},{"cell_type":"markdown","metadata":{"id":"A_W3o-z6J-Sa"},"source":["We will be de novo assembling a SARS-CoV-2 virus (the virus that causes COVID-19), just like the researchers in Wuhan did."]},{"cell_type":"markdown","metadata":{"id":"tiSMqPxRFbdj"},"source":["## Step 1: Examine your reads!\n","\n","For *de novo* assembly, we will be using some [Oxford Nanopore reads](https://www.ncbi.nlm.nih.gov/sra/SRX7615553) of SARS-CoV-2, \n","sequenced by HKU-Shenzhen Hospital."]},{"cell_type":"code","metadata":{"id":"-OGJHkXLseQ5","cellView":"form"},"source":["#@title #### **Exercise: Examine your reads.**\n","%%bash\n","SRR_ID=SRR10948474_OXFORD_NANOPORE\n","less data/${SRR_ID}.fastq | head -n 20 # This shows the first 20 lines of your fastq file. The fastq file is the output from the sequencer."],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"YBjTok_tEtfq"},"source":["### Fields to care about:"]},{"cell_type":"markdown","metadata":{"id":"QpdNF0kyC71R"},"source":[" `@SRR10948474.30344`: This is just the READ_ID, used so we can remember which sequence came from which read.\n","\n","`TGCGATAGTTCAGCACCTGTTTCCCACT....`: This is the string of basepairs that the machine output it sequence that piece of DNA. Oxford nanopore reads can be LONG!\n","\n","`,(\"$+%2$4;E9=?@G?@@'BC.G<8FJA><'N<?@`: This is the read quality. This is code that the machine outputs to describe how confident it was in each of the basepairs it called.  Oxford nanopore reads are often error-prone.\n"]},{"cell_type":"markdown","metadata":{"id":"fZs0fPjjFG3h"},"source":["## Step 2: De novo assemble your genome."]},{"cell_type":"markdown","metadata":{"id":"CwwXZJEu-qf6"},"source":[" This can take up to 10 minutes, so feel free to take a break/ask questions while this runs!"]},{"cell_type":"code","metadata":{"id":"NVAApJPSoj_O","cellView":"form"},"source":["#@title ### **Exercise: De novo assemble your genome**\n","%%bash\n","# IGNORE -- Setting up directory structure/permissions.\n","PATH='/content/software/SPAdes/bin':$PATH\n","DATA_FOLDER=data\n","RESULTS_DIR=results\n","SRR_ID=SRR10948474_OXFORD_NANOPORE\n","\n","# Perform denovo assembly using SPAdes.\n","spades.py -s ${DATA_FOLDER}/${SRR_ID}.fastq -o ${RESULTS_DIR}/${SRR_ID}_spades"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"c6j0GEvVFF9S"},"source":["## Step 3: Examine your contigs."]},{"cell_type":"code","metadata":{"id":"x0gB39xoDo3t","cellView":"form"},"source":["#@title ### **Exercise: Examine your contigs**\n","%%bash \n","SRR_ID=SRR10948474_OXFORD_NANOPORE\n","\n","cp results/${SRR_ID}_spades/contigs.fasta ${SRR_ID}_denovo.fasta\n","head -n 10 results/${SRR_ID}_denovo.fasta  # Show the first 10 lines of the de novo assembled genome."],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"f2EH5hgiD1Y8"},"source":["### Fields to care about.\n"]},{"cell_type":"markdown","metadata":{"id":"rkblI_W6EiQN"},"source":["`NODE_1`: ID of the contig.\n","\n","`length_23792`: The first contig is 23792 basepairs long.\n","\n","`CTATTCTCAGCACCTGTTTCC...`: The assembled sequence.\n"]},{"cell_type":"markdown","metadata":{"id":"JxFdFFSTK52n"},"source":["We see that we assembled *nearly* (about 80%) of the entire SARS-CoV-2 genome!! (SARS-CoV-2 is ~30kbp, we got to ~24kbp). That's pretty good, given a single batch of sequencing reads."]},{"cell_type":"markdown","metadata":{"id":"_H5PtnoHXn1P"},"source":["# Wrapping up"]},{"cell_type":"markdown","metadata":{"id":"0V06OfxFWUy4"},"source":["***Nice job!*** We covered many new things in biology, genomics and machine learning! You've learned concepts that many students don't learn until their PhD!\n","\n","\n","![alt text](https://memeshappen.com/media/created/Job-WELL-done--meme-63067.jpg)\n","\n","As a final exercise, brainstorm a few interesting projects you wish to do in the future! \n"]},{"cell_type":"code","metadata":{"id":"K-mxchSeXeoy","cellView":"form"},"source":["#@title ###**Exercise: What are some other projects you wish to do in the future using biology, genomics, and machine learning?**\n","_1_ = \"\" #@param {type:\"string\"}\n","_2_ = \"\" #@param {type:\"string\"}\n","_3_ = \"\" #@param {type:\"string\"}\n","\n"],"execution_count":null,"outputs":[]}]}